{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3d006a6b",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "from scipy import stats\n",
    "from scipy.stats import linregress\n",
    "from string import ascii_letters\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "sns.set_style(\"whitegrid\")\n",
    "\n",
    "from scipy.stats import ttest_ind\n",
    "from scipy.stats import levene\n",
    "\n",
    "import os\n",
    "import os.path\n",
    "from os import path\n",
    "\n",
    "import glob\n",
    "\n",
    "import math\n",
    "\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "from sklearn.decomposition import PCA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9b9af704",
   "metadata": {},
   "outputs": [],
   "source": [
    "def mean_data(df):\n",
    "    items = list(set([c[:-1] for c in df.columns.to_list()]))\n",
    "\n",
    "    \n",
    "    numbers = list(range(0,len(items)))\n",
    "    data = []\n",
    "    mean = pd.DataFrame(data)\n",
    "    for i in numbers:\n",
    "        cols = [c for c in df if c.startswith(items[i])]\n",
    "\n",
    "        mean[items[i]] = df.loc[:,cols].mean(1)\n",
    "    return mean"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c38b0980",
   "metadata": {},
   "outputs": [],
   "source": [
    "def statistic_f(DATASET1, DATASET2):\n",
    "    #testing normal distribution\n",
    "    if len(DATASET1) > 5000 or len(DATASET2) > 5000:\n",
    "        p_normaldist1 = stats.kstest(DATASET1, stats.norm.cdf)[1]\n",
    "        p_normaldist2 = stats.kstest(DATASET2, stats.norm.cdf)[1]\n",
    "    if len(DATASET1) < 3 or len(DATASET2) < 3:\n",
    "        p_normaldist1 = 1\n",
    "        p_normaldist2 = 1\n",
    "    else:\n",
    "        p_normaldist1 = stats.shapiro(DATASET1)[1]\n",
    "        p_normaldist2 = stats.shapiro(DATASET2)[1]\n",
    "    \n",
    "    #testing variance\n",
    "    stat, p_levene = levene(DATASET1, DATASET2)\n",
    "    \n",
    "    if p_normaldist1 < 0.05 or p_normaldist2 < 0.05:\n",
    "        p_ttest = stats.mannwhitneyu(DATASET1, DATASET2)[1]  #Mann whitney U test\n",
    "        print('The data is not normaly distributed, so the  Mann-Whitney U test was performed')\n",
    "    else:\n",
    "        if p_levene < 0.05:\n",
    "            equal_True_False = False\n",
    "            print('The data is normaly distributed but has no equal variance so Welch´s T-test was perfomred')\n",
    "        else:\n",
    "            equal_True_False = True\n",
    "            print('The data is normaly distributed and has equal variance so Student´s T-test was perfomred')\n",
    "        p_ttest = stats.ttest_ind(DATASET1, DATASET2 ,equal_var=equal_True_False)[1] # t-test\n",
    "    return(p_ttest)\n",
    "\n",
    "def cohens_d(DATASET1, DATASET2):\n",
    "    #calculating Cohen´s d\n",
    "    mean_difference = np.mean(DATASET1) - np.mean(DATASET2)\n",
    "    pooled_standard_deviation = np.sqrt((np.var(DATASET1) + np.var(DATASET2)) / 2)\n",
    "    cohen_d = mean_difference / pooled_standard_deviation\n",
    "    return(cohen_d)\n",
    "\n",
    "def statistic_asterisk(p_value):\n",
    "    #statistics depiction with asterisks\n",
    "    if p_value < 0.0001:\n",
    "        p_value_plot = '****'\n",
    "    elif p_value < 0.001:\n",
    "        p_value_plot = '***'\n",
    "    elif p_value < 0.01:\n",
    "        p_value_plot = '**'\n",
    "    elif p_value < 0.05:\n",
    "        p_value_plot = '*'\n",
    "    else:\n",
    "        p_value_plot = 'ns'\n",
    "    return(p_value_plot)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "483750c3",
   "metadata": {},
   "outputs": [],
   "source": [
    "PATH = 'C:/Users/felix/Documents/Phd'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f1a7cbaa",
   "metadata": {},
   "source": [
    "# Tissue splice analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "75fada95",
   "metadata": {},
   "outputs": [],
   "source": [
    "GO_secretion = pd.read_excel(PATH + '/RNAseq_Felix/GOterm_Lists/GOsecretion_human2.xlsx')\n",
    "GO_secretion = GO_secretion.set_index('geneSymbol')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fb70703f",
   "metadata": {},
   "outputs": [],
   "source": [
    "#read RMATS files and create dictionary with PSI of all splicing events of all conditions(tissues)\n",
    "splicing_dict = {} #create empty dictionary\n",
    "splicing_secretion_dict = {} #create specialized dictionary (GO term secretion) \n",
    "#list of conditions(tissues)\n",
    "list_condition = ['liver', 'bonemarrow', 'brain',  'colon', 'esophagus', 'fat', 'heart', 'kidney'\n",
    "            ,'lung', 'lymphnode', 'placenta', 'skin'\n",
    "            , 'smallintestine', 'spleen', 'stomach', 'testis', 'thyroid'\n",
    "            ,'urinarybladder']\n",
    "source_path = PATH +  '/RNAseq_Felix/human_tissues/Bioinfo/RMATS_filter/'\n",
    "\n",
    "def read_splice_events(TYPE, tissue, path):\n",
    "    if tissue == 'liver':\n",
    "        splicing = pd.read_excel(f'{path}Livervsbrain/{TYPE}_Livervsbrain.xlsx'\n",
    "                                ,usecols = [2,3,4,5,6,7,8,9,10,13,14,17,18,19])\n",
    "        #combine coordinates in one \"coord\" column\n",
    "        splicing['coord_chr'] = splicing.iloc[:, 1]\n",
    "        splicing['coord_1'] = splicing.iloc[:, 3]\n",
    "        splicing['coord_2'] = splicing.iloc[:, 4]\n",
    "        splicing['coord_3'] = splicing.iloc[:, 5]\n",
    "        splicing['coord_4'] = splicing.iloc[:, 6]\n",
    "        splicing['coord_5'] = splicing.iloc[:, 7]\n",
    "        splicing['coord_6'] = splicing.iloc[:, 8]\n",
    "        \n",
    "        coverage_list = []\n",
    "        for i in range(len(splicing)):\n",
    "            coverage_list.append(np.mean(list(map(int, splicing['IJC_SAMPLE_1'][i].split(\",\")))) + np.mean(list(map(int, splicing['SJC_SAMPLE_1'][i].split(\",\")))))\n",
    "        splicing['mean_coverage'] = coverage_list\n",
    "        splicing = splicing.query('mean_coverage > 10')\n",
    "        \n",
    "        #drop the unused columns\n",
    "        splicing = splicing.drop(splicing.columns[[1,3,4,5,6,7,8,9,10,21]],axis = 1)\n",
    "        #include a splice type column\n",
    "        splicing['Type'] = TYPE \n",
    "        return splicing\n",
    "    \n",
    "    else:\n",
    "        \n",
    "        splicing = pd.read_excel(f'{path}Livervs{tissue}/{TYPE}_Livervs{tissue}.xlsx')\n",
    "        #combine coordinates in one \"coord\" column\n",
    "        splicing['coord_chr'] = splicing.iloc[:, 3]\n",
    "        splicing['coord_1'] = splicing.iloc[:, 5]\n",
    "        splicing['coord_2'] = splicing.iloc[:, 6]\n",
    "        splicing['coord_3'] = splicing.iloc[:, 7]\n",
    "        splicing['coord_4'] = splicing.iloc[:, 8]\n",
    "        splicing['coord_5'] = splicing.iloc[:, 9]\n",
    "        splicing['coord_6'] = splicing.iloc[:, 10]\n",
    "        \n",
    "        coverage_list = []\n",
    "        for i in range(len(splicing)):\n",
    "            coverage_list.append(np.mean(list(map(int, splicing['IJC_SAMPLE_1'][i].split(\",\")))) + np.mean(list(map(int, splicing['SJC_SAMPLE_1'][i].split(\",\")))))\n",
    "        splicing['mean_coverage'] = coverage_list\n",
    "        splicing = splicing.query('mean_coverage > 10')\n",
    "        \n",
    "        #drop the unused columns\n",
    "        splicing = splicing.drop(splicing.columns[[0,1, 3, 5,6,7,8,9,10,11,12,13,14,15,16,17,18,19]],axis = 1)\n",
    "        splicing = splicing.drop('mean_coverage',axis = 1)\n",
    "        #include a splice type column\n",
    "        splicing['Type'] = TYPE \n",
    "        return splicing\n",
    "\n",
    "\n",
    "#loop over condition in \n",
    "for x in list_condition:\n",
    "    #read files\n",
    "    SE = read_splice_events('SE', x, source_path)\n",
    "    RI = read_splice_events('RI', x, source_path)\n",
    "    A3SS = read_splice_events('A3SS', x, source_path)\n",
    "    A5SS = read_splice_events('A5SS', x, source_path)\n",
    "    \n",
    "    #add to the dictionary a DataFrame with all splice events in the key of the condition \n",
    "    splicing_dict[f'{x}'] = pd.DataFrame(pd.concat([SE, RI,  A5SS, A3SS], axis = 0))\n",
    "    #add to the specialized dicitonary with the merge with the secretion GO term\n",
    "    splicing_secretion_dict[f'{x}'] = splicing_dict[f'{x}'].reset_index().merge(GO_secretion\n",
    "                                        , on = 'geneSymbol')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ce1441d1",
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in splicing_dict:\n",
    "    splicing_dict[i] = splicing_dict[i].set_index(['geneSymbol', 'strand', 'coord_chr', 'coord_1', 'coord_2', 'coord_3'\n",
    "                                                            , 'coord_4', 'coord_5', 'coord_6', 'Type'])\n",
    "    \n",
    "for i in splicing_secretion_dict:\n",
    "    splicing_secretion_dict[i] = splicing_secretion_dict[i].drop(['index', 'GO_term'], axis = 1)\n",
    "    splicing_secretion_dict[i] = splicing_secretion_dict[i].set_index(['geneSymbol', 'strand', 'coord_chr', 'coord_1' \n",
    "                                                            ,'coord_2', 'coord_3'\n",
    "                                                            , 'coord_4', 'coord_5', 'coord_6', 'Type'])\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c9685e46",
   "metadata": {},
   "outputs": [],
   "source": [
    "splicing = splicing_dict['liver']\n",
    "for i in splicing_dict:\n",
    "    if i == 'liver':\n",
    "        pass\n",
    "    else: \n",
    "        splicing = splicing.merge(splicing_dict[i], left_index=True, right_index=True, how = 'inner')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "43adb04f",
   "metadata": {},
   "outputs": [],
   "source": [
    "splicing_secretion = splicing_secretion_dict['liver']\n",
    "for i in splicing_secretion_dict:\n",
    "    if i == 'liver':\n",
    "        pass\n",
    "    else: \n",
    "        splicing_secretion = splicing_secretion.merge(splicing_secretion_dict[i], \n",
    "                                                      left_index=True, right_index=True, how = 'inner')\n",
    "     "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "59fc5d9c",
   "metadata": {},
   "outputs": [],
   "source": [
    "splicing.reset_index().sort_values(by=['geneSymbol']).reset_index().drop('index' \n",
    "                                , axis = 1).to_excel(PATH + 'Projects/SEC31A_Exon24c/splicing.xlsx')\n",
    "splicing_secretion.reset_index().sort_values(by=['geneSymbol']).reset_index().drop('index'\n",
    "                                , axis = 1).to_excel(PATH + 'Projects/SEC31A_Exon24c/splicing_secretion.xlsx')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "23cadc4e",
   "metadata": {},
   "outputs": [],
   "source": [
    "splicing = pd.read_excel(PATH + '/Projects/SEC31A_Exon24c/splicing.xlsx', index_col = [1,2,3,4,5,6,7,8,9,10])\n",
    "splicing = splicing.drop('Unnamed: 0', axis = 1)\n",
    "splicing_secretion = pd.read_excel(PATH +  '/Projects/SEC31A_Exon24c/splicing_secretion.xlsx', index_col = [1,2,3,4,5,6,7,8,9,10])\n",
    "splicing_secretion = splicing_secretion.drop('Unnamed: 0', axis = 1)\n",
    "splicing = splicing.fillna(0)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "417a7dda",
   "metadata": {},
   "source": [
    "# Pricnipal component analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b88e9f00",
   "metadata": {},
   "outputs": [],
   "source": [
    "#preparing and transposing splicing data for PCA \n",
    "plot = splicing.loc[(splicing).any(axis=1)]\n",
    "plot_transposed = plot.transpose()\n",
    "plot_transposed.reset_index(inplace=True)\n",
    "\n",
    "# Separating out the tarplott\n",
    "y = plot_transposed.loc[:,['index']].values\n",
    "\n",
    "plot_transposed = plot_transposed.drop(['index'], axis=1)\n",
    "features = list(plot_transposed.columns)\n",
    "x = plot_transposed.loc[:, features].values\n",
    "\n",
    "#Standardizing the features\n",
    "x = StandardScaler().fit_transform(x)\n",
    "\n",
    "pca = PCA(n_components=2)\n",
    "principalComponents = pca.fit_transform(x)\n",
    "principalDf = pd.DataFrame(data = principalComponents\n",
    "             , columns = ['principal component 1', 'principal component 2'])\n",
    "\n",
    "plot_transposed = plot.transpose()\n",
    "plot_transposed.reset_index(inplace=True)\n",
    "\n",
    "finalDf = pd.concat([principalDf, plot_transposed[['index']]], axis = 1)\n",
    "\n",
    "finalDf.columns = ['principal component 1', 'principal component 2', 'index']\n",
    "finalDf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c1a7f220",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Finalising data and correct index\n",
    "finalDf_split = finalDf['index'].str.split('_', n = 1, expand = True)\n",
    "finalDf_split.columns = [ 'b' , 'index']\n",
    "finalDf_split = finalDf_split['index'].str.split('_', n = 1, expand = True)\n",
    "finalDf_split.columns = ['index', 'b']\n",
    "\n",
    "#Add a column PSI24c_mean, as we wanted to preserve the color code for the tissues in all plots\n",
    "finalDf['PSI24c_mean'] = list_PSI24c_mean\n",
    "finalDf = finalDf.sort_values(by = 'PSI24c_mean', ascending = False)\n",
    "\n",
    "#correct the index\n",
    "finalDf['index'] = finalDf_split['index']\n",
    "finalDf = finalDf.replace('Liver', 'liver')\n",
    "finalDf = finalDf.replace('Bonemarrow', 'bone marrow')\n",
    "finalDf = finalDf.replace('smallintestine', 'small intestine')\n",
    "finalDf = finalDf.replace('urinarybladder', 'urinary bladder')\n",
    "finalDf = finalDf.replace('lymphnode', 'lymph node')\n",
    "finalDf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "be4558a0",
   "metadata": {},
   "outputs": [],
   "source": [
    "#draw PCA plot\n",
    "sns.scatterplot(data = finalDf, x = 'principal component 1', y = 'principal component 2', hue = 'index' \n",
    "                , palette = sns.color_palette(\"hls\", 22)\n",
    "               , style = 'index')\n",
    "plt.legend(bbox_to_anchor=(1.02, 1), loc='upper left', borderaxespad=0)\n",
    "plt.xlim([-25, 20])\n",
    "plt.ylim([-40, 150])\n",
    "#plt.savefig(PATH + \"Projects/SEC31A_Exon24c/Figures/TissueData_splicing_PCA.png\", dpi = 600, bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "657b8cb4",
   "metadata": {},
   "outputs": [],
   "source": [
    "#draw zoomed in PCA plot\n",
    "sns.scatterplot(data = finalDf, x = 'principal component 1', y = 'principal component 2', hue = 'index' \n",
    "                , palette = sns.color_palette(\"hls\", 22)\n",
    "               , style = 'index')\n",
    "plt.legend(bbox_to_anchor=(1.02, 1), loc='upper left', borderaxespad=0)\n",
    "plt.xlim([-20, -5])\n",
    "plt.ylim([-20, 15])\n",
    "#plt.savefig(PATH + \"Projects/SEC31A_Exon24c/Figures/TissueData_splicing_PCA_zoom.png\", dpi = 600, bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "984612ee",
   "metadata": {},
   "source": [
    "# Lollipop plot"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "763cdeb7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Create a dataframe with average PSI of all secretory related splice events ofer the tissues\n",
    "splicing_secretion_lolliplot = mean_data(splicing_secretion)\n",
    "splicing_secretion_lolliplot = splicing_secretion_lolliplot.drop('PSI_lymphnode_1', axis = 1)\n",
    "\n",
    "#creating and delta column for sorting by the maximum difference\n",
    "splicing_secretion_lolliplot['max'] = splicing_secretion_lolliplot.max(axis=1)\n",
    "splicing_secretion_lolliplot['min'] = splicing_secretion_lolliplot.min(axis=1)\n",
    "splicing_secretion_lolliplot['delta'] = splicing_secretion_lolliplot['max'] - splicing_secretion_lolliplot['min']\n",
    "splicing_secretion_lolliplot = splicing_secretion_lolliplot.sort_values(by = 'delta', ascending = False)\n",
    "#filtering by delta PSI of >0.5, <-0.5\n",
    "splicing_secretion_lolliplot = splicing_secretion_lolliplot.query('delta > .5 | delta < -.5')\n",
    "\n",
    "splicing_secretion_lolliplot = splicing_secretion_lolliplot.rename(columns={\"PSI_kidney_\": \"kidney\", \"PSI_lung_\": \"lung\"\n",
    "                , 'PSI_skin_': 'skin', 'PSI_testis_' : 'testis', 'PSI_urinarybladder_' : 'urinarybladder'\n",
    "                , 'PSI_spleen_' : 'spleen', 'PSI_Liver_' : 'liver', 'PSI_smallintestine_' : 'smallintestine'\n",
    "                , 'PSI_fat_' : 'fat', 'PSI_thyroid_' : 'thyroid', 'PSI_esophagus_' : 'esophagus'\n",
    "                , 'PSI_brain_' : 'brain', 'PSI_heart_' : 'heart', 'PSI_lymphnode_' : 'lymphnode'\n",
    "                , 'PSI_placenta_' : 'placenta', 'PSI_stomach_' : 'stomach', 'PSI_Bonemarrow_' : 'bonemarrow'\n",
    "                , 'PSI_colon_' : 'colon'})\n",
    "splicing_secretion_lolliplot = splicing_secretion_lolliplot.fillna(0)\n",
    "splicing_secretion_lolliplot = splicing_secretion_lolliplot.reset_index()\n",
    "splicing_secretion_lolliplot = splicing_secretion_lolliplot.drop_duplicates(subset='coord_1')\n",
    "splicing_secretion_lolliplot['coords'] =  splicing_secretion_lolliplot['geneSymbol'] + '  ' + splicing_secretion_lolliplot['coord_chr'] + ': ' + splicing_secretion_lolliplot['coord_1'].astype(str) + '-'  + splicing_secretion_lolliplot['coord_2'].astype(str)\n",
    "splicing_secretion_lolliplot = splicing_secretion_lolliplot.drop(['geneSymbol', 'strand', 'coord_chr', 'coord_1', 'coord_2',\n",
    "                                                                  'coord_3', 'coord_4', 'coord_5', 'coord_6', \n",
    "                                                                  'Type', 'liver', 'lymphnode', 'brain', 'bonemarrow', \n",
    "                                                                  'testis', 'kidney', 'skin', 'stomach', 'colon',\n",
    "                                                                 'placenta', 'spleen', 'esophagus', 'thyroid', 'lung',\n",
    "                                                                  'urinarybladder', 'fat', 'smallintestine', 'heart'\n",
    "                                                                 ],\n",
    "                                                                 axis = 1)\n",
    " \n",
    "# Reorder it following the values of the first value:\n",
    "splicing_secretion_lolliplot_ordered = splicing_secretion_lolliplot.sort_values(by='delta')\n",
    "my_range=range(1,len(splicing_secretion_lolliplot.index)+1)\n",
    " \n",
    "palette_generator = sns.color_palette(\"PiYG_r\", 10)\n",
    "palette_generator2 =  sns.color_palette(\"cividis\", 5)\n",
    "# The horizontal plot is made using the hline function\n",
    "plt.figure(figsize=(8,8))\n",
    "plt.hlines(y=my_range, xmin=splicing_secretion_lolliplot_ordered['min'],\n",
    "           xmax=splicing_secretion_lolliplot_ordered['max']\n",
    "           , color='grey'\n",
    "           ,alpha=0.8, linewidths = 3)\n",
    "\n",
    "plt.scatter(splicing_secretion_lolliplot_ordered['min'], my_range, color=palette_generator2[4], alpha=1, label='Delta PSI'\n",
    "           ,s = 100)\n",
    "plt.scatter(splicing_secretion_lolliplot_ordered['min'], my_range, color=palette_generator[0], alpha=1, label='Min PSI'\n",
    "           ,s = 100)\n",
    "plt.scatter(splicing_secretion_lolliplot_ordered['max'], my_range, color=palette_generator[9], alpha=1 , label='Max PSI'\n",
    "           ,s = 100)\n",
    "plt.legend()\n",
    "plt.legend(bbox_to_anchor=(1.02, 1), loc='upper left', borderaxespad=0)\n",
    "# Add title and axis names\n",
    "plt.yticks(my_range, splicing_secretion_lolliplot_ordered['coords'])\n",
    "plt.xlabel('PSI')\n",
    "\n",
    "# Show the graph\n",
    "#plt.savefig(PATH + \"Projects/SEC31A_Exon24c/Figures/Tissue_secretion_splicing_MINMAX.png\", dpi = 600, bbox_inches = \"tight\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5c8880d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Reorder it following the values of the first value:\n",
    "splicing_secretion_lolliplot_ordered = splicing_secretion_lolliplot.sort_values(by='delta')\n",
    "my_range=range(1,len(splicing_secretion_lolliplot.index)+1)\n",
    " \n",
    "palette_generator = sns.color_palette(\"PiYG_r\", 10)\n",
    "palette_generator2 =  sns.color_palette(\"cividis\", 5)\n",
    "# The horizontal plot is made using the hline function\n",
    "plt.figure(figsize=(3,8))\n",
    "#plt.legend()\n",
    "   \n",
    "plt.scatter(splicing_secretion_lolliplot_ordered['delta'], my_range, color=palette_generator2[4], alpha=1, label='Delta PSI'\n",
    "           ,s = 100)\n",
    "# Add title and axis names\n",
    "plt.yticks(my_range, splicing_secretion_lolliplot_ordered['coords'])\n",
    "plt.xticks((0.5, 0.75, 1.0))\n",
    "#plt.title(\"Comparison of the lowest and the highest PSI per exon\", loc='left')\n",
    "plt.xlabel('Delta PSI')\n",
    "plt.xlim(0.4,1.05)\n",
    "#plt.ylabel('coords')\n",
    "\n",
    "# Show the graph\n",
    "#plt.savefig(PATH + \"Projects/SEC31A_Exon24c/Figures/Tissue_secretion_splicing_DELTA.png\", dpi = 600, bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1d9cfde9",
   "metadata": {},
   "source": [
    "# Exon24c Tissue specific splicing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6168a536",
   "metadata": {},
   "outputs": [],
   "source": [
    "#extract SEC31A exon 24c of the global splicing list\n",
    "splicing_E24c  = splicing_secretion.loc['SEC31A', '-', 'chr4', 82830936, 82830975, 82828999, 82829058]\n",
    "splicing_E24c['exon'] = ['E24c_1', 'E24c_3', 'PSI SEC31A Exon 24c']\n",
    "splicing_E24c = splicing_E24c.reset_index()\n",
    "splicing_E24c = splicing_E24c.set_index('exon')\n",
    "splicing_E24c = splicing_E24c.drop(['coord_5', 'coord_6', 'Type'], axis = 1)\n",
    "splicing_E24c = splicing_E24c.transpose()\n",
    "splicing_E24c = splicing_E24c.reset_index()\n",
    "splicing_E24c.columns = ['tissue','E24c_1', 'E24c_3', 'PSI SEC31A Exon 24c']\n",
    "\n",
    "# remove names from sinlge samples, and replace it with an index of the tissues\n",
    "string_list = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'PSI_', '_']\n",
    "for x in string_list:\n",
    "    splicing_E24c['tissue'] = splicing_E24c['tissue'].str.replace(x, '')\n",
    "\n",
    "# homogenize the index    \n",
    "splicing_E24c = splicing_E24c.replace('Liver', 'liver')\n",
    "splicing_E24c = splicing_E24c.replace('Bonemarrow', 'bone marrow')\n",
    "splicing_E24c = splicing_E24c.replace('smallintestine', 'small intestine')\n",
    "splicing_E24c = splicing_E24c.replace('urinarybladder', 'urinary bladder')\n",
    "splicing_E24c = splicing_E24c.replace('lymphnode', 'lymph node')\n",
    "\n",
    "splicing_E24c = splicing_E24c.set_index('tissue')\n",
    "\n",
    "#create column to sort the samples from mean maximum to mean minimum\n",
    "list_PSI24c_mean = []\n",
    "for i in splicing_E24c.transpose():\n",
    "    list_PSI24c_mean.append(np.mean(splicing_E24c.loc[i]['PSI SEC31A Exon 24c']))\n",
    "splicing_E24c['PSI24c_mean'] = list_PSI24c_mean\n",
    "splicing_E24c = splicing_E24c.sort_values(by = 'PSI24c_mean', ascending = False)\n",
    "\n",
    "\n",
    "# plot the bar plot\n",
    "palette_generation = sns.color_palette(\"hls\", 22)\n",
    "sns.barplot(data=splicing_E24c.reset_index()\n",
    "            , x=\"tissue\", y=\"PSI SEC31A Exon 24c\", ci = \"sd\", palette=palette_generation\n",
    "           , linewidth=2, edgecolor=\"0\"\n",
    "           , errorbar=(\"pi\", 50), capsize=.3, errcolor=\"0\")\n",
    "plt.tick_params(axis='x', rotation=90)\n",
    "#plt.savefig(PATH + \"Projects/SEC31A_Exon24c/Figures/SEC31A_E24c_splicing.png\", dpi = 600, bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7f7eaabb",
   "metadata": {},
   "source": [
    "# Mouse tissue validation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f4e048ec",
   "metadata": {},
   "outputs": [],
   "source": [
    "#create DataFrame of the data from the quantification of the radioactive gel\n",
    "E24csplicing_mousetissues = pd.DataFrame({'PSI 24c' : [ 0.703915241, 0.718047615, 0.710920947\n",
    "                                                        , 0.432203698, 0.386756401, 0.420612286\n",
    "                                                        , 0.09155929, 0.097797553, 0.151566682\n",
    "                                                        , 0.040899208, 0.028932168, 0.000193505\n",
    "                                                        , 0.145476394, 0.183124488, 0.22928863\n",
    "                                                        , 0.020357807, 0.046647126, 0.016379024\n",
    "                                                        ],\n",
    "                                           'tissue' : ['liver', 'liver','liver',\n",
    "                                                     'kidney', 'kidney', 'kidney',\n",
    "                                                    'spleen', 'spleen', 'spleen',\n",
    "                                                    'lung', 'lung', 'lung',\n",
    "                                                    'heart', 'heart', 'heart',\n",
    "                                                    'brain', 'brain', 'brain',\n",
    "                                                    ]\n",
    "                                         })\n",
    "E24csplicing_mousetissues = E24csplicing_mousetissues.set_index('tissue')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f9ca3a1a",
   "metadata": {},
   "outputs": [],
   "source": [
    "#create a DataFrame assigning a color for each tissue, to make the color code consistent\n",
    "palette_generation = sns.color_palette(\"hls\", 22)\n",
    "palette_dataframe = pd.DataFrame({'tissue' :  ['liver', 'kidney', 'stomach', 'thyroid', 'colon', 'smallintestine',\n",
    "                                                    'bonemarrow', 'lung', 'placenta', \n",
    "                                                   'urinarybladder', 'esophagus','skin',\n",
    "                                                   'testis', 'lymphnode', 'spleen','fat', \n",
    "                                                   'heart', 'brain'],\n",
    "                                'color' : palette_generation[0:18]})\n",
    "palette_dataframe = palette_dataframe.set_index('tissue')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "565d952b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# draw barplot\n",
    "palette = [palette_dataframe.loc['liver'][0], palette_dataframe.loc['kidney'][0], palette_dataframe.loc['spleen'][0]\n",
    "          ,palette_dataframe.loc['lung'][0], palette_dataframe.loc['heart'][0], palette_dataframe.loc['brain'][0]]\n",
    "sns.barplot(data=E24csplicing_mousetissues.reset_index(), x=\"tissue\", y=\"PSI 24c\", ci = \"sd\", palette=palette\n",
    "           , linewidth=2, edgecolor=\"0\"\n",
    "           , errorbar=(\"pi\", 50), capsize=.3, errcolor=\"0\")\n",
    "plt.tick_params(axis='x', rotation=90)\n",
    "#plt.savefig(PATH + \"Projects/SEC31A_Exon24c/Figures/SEC31A_E24c_splicing_mouse_validation.png\", dpi = 600, bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7b7c4e72",
   "metadata": {},
   "outputs": [],
   "source": [
    "#preparing data for linear regression between mouse and human data\n",
    "tissue_list = ['liver', 'kidney', 'spleen', 'lung', 'heart', 'brain']\n",
    "mouse_mean = []\n",
    "human_mean = []\n",
    "mouse_std = []\n",
    "human_std = []\n",
    "#creating lists for the mean and standart deviasion of exon 24c PSI in mouse and human data\n",
    "for i in tissue_list:\n",
    "    mouse_mean.append(np.mean(E24csplicing_mousetissues.loc[i]['PSI 24c']))\n",
    "    mouse_std.append(np.std(E24csplicing_mousetissues.loc[i]['PSI 24c']))\n",
    "    human_mean.append(np.mean(splicing_E24c.loc[i]['PSI SEC31A Exon 24c']))\n",
    "    human_std.append(np.std(splicing_E24c.loc[i]['PSI SEC31A Exon 24c']))\n",
    "\n",
    "# creating DataFrame of the mouse and human data\n",
    "E24csplicing_validationplot_mouse = pd.DataFrame({'PSI Exon 24c mouse RT-qPCR' : mouse_mean,\n",
    "                                                 'mouse_std' : mouse_std,\n",
    "                                                'tissue' : tissue_list\n",
    "                                                 })    \n",
    "E24csplicing_validationplot_human = pd.DataFrame({'PSI Exon 24c human RNAseq data' : human_mean,\n",
    "                                                 'human_std' : human_std \n",
    "                                                 })    \n",
    "E24csplicing_validationplot = pd.concat([E24csplicing_validationplot_human, E24csplicing_validationplot_mouse]\n",
    "                                       ,axis = 1)\n",
    "#plot the y and x error bars in propper color \n",
    "for n in range(0,6):\n",
    "    plt.errorbar(human_mean[n], mouse_mean[n], xerr = human_std[n], yerr=mouse_std[n]\n",
    "             , fmt=\"none\"\n",
    "            ,color = palette[n]\n",
    "             ,capsize = 3\n",
    "             ,alpha= .5\n",
    "            )\n",
    "    \n",
    "#plot the mean values\n",
    "sns.scatterplot(data = E24csplicing_validationplot, x = 'PSI Exon 24c human RNAseq data'\n",
    "                , y = 'PSI Exon 24c mouse RT-qPCR'\n",
    "                , hue = 'tissue'\n",
    "                , palette = palette) \n",
    "\n",
    "#plot the regression line\n",
    "sns.regplot(data = E24csplicing_validationplot, x = 'PSI Exon 24c human RNAseq data'\n",
    "                , y = 'PSI Exon 24c mouse RT-qPCR'\n",
    "           , marker = ''\n",
    "           , ci = 0\n",
    "           , color = 'gray'\n",
    "           )\n",
    "\n",
    "plt.legend(bbox_to_anchor=(1.02, 1), loc='upper left', borderaxespad=0)\n",
    "\n",
    "reg_stats = linregress(E24csplicing_validationplot['PSI Exon 24c human RNAseq data']\n",
    "                 , E24csplicing_validationplot['PSI Exon 24c mouse RT-qPCR'])\n",
    "\n",
    "#plt.savefig(PATH + \"Projects/SEC31A_Exon24c/Figures/SEC31A_E24c_splicing_mouse_validation_reg.png\", dpi = 600, bbox_inches = \"tight\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fb85a1ad",
   "metadata": {},
   "outputs": [],
   "source": [
    "linregress(E24csplicing_validationplot['PSI Exon 24c human RNAseq data']\n",
    "                 , E24csplicing_validationplot['PSI Exon 24c mouse RT-qPCR'])"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "56614231",
   "metadata": {},
   "source": [
    "# Correlation with Exon24c"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "88e1ebda",
   "metadata": {},
   "outputs": [],
   "source": [
    "#preparing exon 24c alternative splicing data to mach the gene expression DataFrame\n",
    "splicing_E24c  = splicing_secretion.loc['SEC31A', '-', 'chr4', 82830936, 82830975, 82828999, 82829058, 82842184, 82842481]\n",
    "splicing_E24c = splicing_E24c.drop(['PSI_Bonemarrow_3', 'PSI_Bonemarrow_7', 'PSI_colon_4','PSI_heart_5', 'PSI_lymphnode_7'\n",
    "                                   ,'PSI_lymphnode_14', 'PSI_testis_3'\n",
    "                                    , 'PSI_thyroid_6', 'PSI_lung_7']\n",
    "                                  , axis = 1)\n",
    "splicing_E24c = splicing_E24c.transpose()#.reset_index().sort_values(by = 'index').set_index('index')\n",
    "splicing_E24c.columns = ['SEC31A Exon 24c']\n",
    "splicing_E24c = splicing_E24c.transpose()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c8016a18",
   "metadata": {},
   "outputs": [],
   "source": [
    "#read in DEseq2 DataFrame with gene counts of all tissues \n",
    "genecounts_all = pd.read_csv(PATH + '/RNAseq_Felix/human_tissues/Bioinfo/Deseq/dataframeall.csv', sep=',', index_col=0)\n",
    "genecounts_all = genecounts_all.reset_index()\n",
    "genecounts_all = genecounts_all.drop(['liver4', 'liver5', 'bonemarrow8', 'colon6', 'lung7','heart5', 'lymphnode'\n",
    "                                      , 'ovary1', 'ovary2', 'ovary3'\n",
    "                                      , 'ovary4',  'bonemarrow9', 'testis3'\n",
    "                                     ], axis = 1)\n",
    "genecounts_all = genecounts_all.set_index(['index'])\n",
    "genecounts_all = genecounts_all.transpose().reset_index().set_index('index').transpose()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8ef223a8",
   "metadata": {},
   "outputs": [],
   "source": [
    "#merging SEC31A splicing PSI with gene counts\n",
    "splicing_E24c.columns = genecounts_all.columns\n",
    "genecounts_all_Exon24 = pd.concat([splicing_E24c, genecounts_all])\n",
    "genecounts_all_Exon24"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "06e8ae56",
   "metadata": {},
   "outputs": [],
   "source": [
    "#calculating the pearson correlation between the PSI and Exon 24a and all other genes in all tissues\n",
    "GE_all = genecounts_all_Exon24\n",
    "corr_list = []\n",
    "for i in GE_all.reset_index()['index']:\n",
    "    Corr_df = pd.DataFrame([genecounts_all_Exon24.loc['SEC31A Exon 24c'], genecounts_all_Exon24.loc[f'{i}']])\n",
    "    #print(Corr_df.transpose().corr()[f'A3SS'][f'{i}'])\n",
    "    corr_list.append(Corr_df.transpose().corr(method='pearson')['SEC31A Exon 24c'][f'{i}'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e9c0a5d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "Correlation = pd.DataFrame(corr_list)\n",
    "Correlation['Gene_ID'] = GE_all.reset_index()['index']\n",
    "Correlation.columns = ['pearson_24c', 'GENE_ID']\n",
    "Correlation = Correlation.set_index('GENE_ID')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dcfc6753",
   "metadata": {},
   "outputs": [],
   "source": [
    "#calculating the spearman correlation between the PSI and Exon 24a and all other genes in all tissues\n",
    "GE_all = genecounts_all_Exon24\n",
    "corr_list = []\n",
    "for i in GE_all.reset_index()['index']:\n",
    "    Corr_df = pd.DataFrame([genecounts_all_Exon24.loc['SEC31A Exon 24c'], genecounts_all_Exon24.loc[f'{i}']])\n",
    "    #print(Corr_df.transpose().corr()[f'A3SS'][f'{i}'])\n",
    "    corr_list.append(Corr_df.transpose().corr(method = 'spearman')['SEC31A Exon 24c'][f'{i}'])\n",
    "    \n",
    "Correlation['spearman_24c'] = corr_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8ca6d6c1",
   "metadata": {},
   "outputs": [],
   "source": [
    "Correlation = Correlation.drop('SEC31A Exon 24c')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d8e111bf",
   "metadata": {},
   "outputs": [],
   "source": [
    "Correlation.columns = ['Pearson_correlation', 'Spearman_correlation']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2e0fa525",
   "metadata": {},
   "outputs": [],
   "source": [
    "Correlation = Correlation.reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "641925e9",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Correlation.to_excel(PATH + '/Projects/SEC31A_Exon24c/Correlation.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "39ae9f1f",
   "metadata": {},
   "outputs": [],
   "source": [
    "#plotting the correlation with highlighted GO terms\n",
    "\n",
    "# loading the correltaion list\n",
    "Correlation = pd.read_excel(PATH + '/Projects/SEC31A_Exon24c/Correlation.xlsx', index_col= 0)\n",
    "Correlation = Correlation.sort_index(ascending=True)\n",
    "\n",
    "#loading various GO temr lists \n",
    "Collagen_Chylomicrons = pd.read_excel(PATH + '/RNAseq_Felix/GOterm_Lists/Collagen_Chylomicrons.xlsx')\n",
    "\n",
    "GOsecretion_human2 = pd.read_excel(PATH + '/RNAseq_Felix/GOterm_Lists/GOsecretion_human2.xlsx')\n",
    "GOsecretion_human2.columns = ['GENE_ID', 'GO_term']\n",
    "\n",
    "GO_term_RNA_splicing = pd.read_excel(PATH + '/RNAseq_Felix/GOterm_Lists/GO_term_RNA_splicing.xlsx')\n",
    "GO_term_RNA_splicing.columns = ['GENE_ID', 'GO_term']\n",
    "\n",
    "GO_terms = pd.concat([GOsecretion_human2, GO_term_RNA_splicing, Collagen_Chylomicrons])\n",
    "GO_terms = GO_terms.drop_duplicates(subset = 'GENE_ID')\n",
    "\n",
    "#merge Correlation list and GO term lists\n",
    "Correlation = Correlation.merge(GO_terms, on = 'GENE_ID', how = 'outer')\n",
    "Correlation = Correlation.dropna(subset=['Pearson_correlation', 'Spearman_correlation'])\n",
    "Correlation = Correlation.fillna('other')\n",
    "\n",
    "Correlation = Correlation.reset_index().drop('index', axis = 1)\n",
    "\n",
    "#create various color palettes\n",
    "palette_generation_2 = sns.color_palette(\"PiYG_r\", 20)\n",
    "palette3 = sns.color_palette(\"Dark2\", 4)\n",
    "\n",
    "#plot each GO term in a scatter plot\n",
    "plt.figure(figsize=(10,10))\n",
    "plt.scatter(data = Correlation.query('GO_term == \"other\"'), x = 'Pearson_correlation', y = 'Spearman_correlation'\n",
    "           , color = 'grey', alpha = .4, label = 'other')\n",
    "plt.scatter(data = Correlation.query('GO_term == \"secretion\"'), x = 'Pearson_correlation', y = 'Spearman_correlation'\n",
    "           , color = palette3[0], alpha = .9, label = 'secretion')\n",
    "plt.scatter(data = Correlation.query('GO_term == \"mRNA splicing\"'), x = 'Pearson_correlation', y = 'Spearman_correlation'\n",
    "           , color = palette3[1], alpha = .9, label = 'mRNA splicing')\n",
    "\n",
    "plt.scatter(data = Correlation.query('GO_term == \"Collagens\"'), x = 'Pearson_correlation', y = 'Spearman_correlation'\n",
    "           , color = palette3[2], alpha = .9, label = 'Collagens')\n",
    "plt.scatter(data = Correlation.query('GO_term == \"Chylomicrons\"'), x = 'Pearson_correlation', y = 'Spearman_correlation'\n",
    "           , color = palette3[3], alpha = .9, label = 'Chylomicrons')\n",
    "\n",
    "#plot legend\n",
    "plt.legend(loc='best', bbox_to_anchor=(0.5, 0., 0.5, 0.5))\n",
    "plt.legend(title='GO terms')\n",
    "\n",
    "#mark highlighted genes with error and gene name\n",
    "text = Correlation['GENE_ID']\n",
    "for i, label in enumerate(text):\n",
    "    if text[i] == 'RBM47':\n",
    "        plt.annotate(text[i],  xy = (Correlation['Pearson_correlation'][i]+0.00, Correlation['Spearman_correlation'][i])\n",
    "                     , xytext = (Correlation['Pearson_correlation'][i]+.1, Correlation['Spearman_correlation'][i]+0.1)\n",
    "                     , size=15, color = palette3[1],\n",
    "                    arrowprops=dict(facecolor=palette3[1]\n",
    "                                    , shrink=0.05\n",
    "                                    , alpha = .8\n",
    "                                   ))\n",
    "    if text[i] == 'MIA2':\n",
    "        plt.annotate(text[i],  xy = (Correlation['Pearson_correlation'][i]+0.00, Correlation['Spearman_correlation'][i])\n",
    "                     , xytext = (Correlation['Pearson_correlation'][i]+.1, Correlation['Spearman_correlation'][i]-.15)\n",
    "                     , size=15, color = palette3[0],\n",
    "                    arrowprops=dict(facecolor=palette3[0]\n",
    "                                    , shrink=0.05\n",
    "                                    , alpha = .8\n",
    "                                   ))\n",
    "        \n",
    "    if text[i] == 'LSR':\n",
    "        plt.annotate(text[i],  xy = (Correlation['Pearson_correlation'][i]+0.005, Correlation['Spearman_correlation'][i])\n",
    "                     , xytext = (Correlation['Pearson_correlation'][i]+.1, Correlation['Spearman_correlation'][i]-.15)\n",
    "                     , size=15, color = palette3[3],\n",
    "                    arrowprops=dict(facecolor=palette3[3]\n",
    "                                    , shrink=0.05\n",
    "                                    , alpha = .8\n",
    "                                   ))\n",
    "    if text[i] == 'APOB':\n",
    "        plt.annotate(text[i],  xy = (Correlation['Pearson_correlation'][i]+0.005, Correlation['Spearman_correlation'][i])\n",
    "                     , xytext = (Correlation['Pearson_correlation'][i]+.12, Correlation['Spearman_correlation'][i]-.1)\n",
    "                     , size=15, color = palette3[3],\n",
    "                    arrowprops=dict(facecolor=palette3[3]\n",
    "                                    , shrink=0.05\n",
    "                                    , alpha = .8\n",
    "                                   ))\n",
    "    if text[i] == 'APOA1':\n",
    "        plt.annotate(text[i],  xy = (Correlation['Pearson_correlation'][i]+0.005, Correlation['Spearman_correlation'][i])\n",
    "                     , xytext = (Correlation['Pearson_correlation'][i]+.12, Correlation['Spearman_correlation'][i]-.1)\n",
    "                     , size=15, color = palette3[3],\n",
    "                    arrowprops=dict(facecolor=palette3[3]\n",
    "                                    , shrink=0.05\n",
    "                                    , alpha = .8\n",
    "                                   ))\n",
    "    if text[i] == 'APOH':\n",
    "        plt.annotate(text[i],  xy = (Correlation['Pearson_correlation'][i]+0.005, Correlation['Spearman_correlation'][i])\n",
    "                     , xytext = (Correlation['Pearson_correlation'][i]-.2, Correlation['Spearman_correlation'][i]+.15)\n",
    "                     , size=15, color = palette3[3],\n",
    "                    arrowprops=dict(facecolor=palette3[3]\n",
    "                                    , shrink=0.05\n",
    "                                    , alpha = .8\n",
    "                                   ))\n",
    "        \n",
    "    if text[i] == 'COL1A1':\n",
    "        plt.annotate(text[i],  xy = (Correlation['Pearson_correlation'][i]+0.000, Correlation['Spearman_correlation'][i])\n",
    "                     , xytext = (Correlation['Pearson_correlation'][i]+.2, Correlation['Spearman_correlation'][i]-.2)\n",
    "                     , size=15, color = palette3[2],\n",
    "                    arrowprops=dict(facecolor=palette3[2]\n",
    "                                    , shrink=0.05\n",
    "                                    , alpha = .8\n",
    "                                   ))\n",
    "        \n",
    "    if text[i] == 'COL1A2':\n",
    "        plt.annotate(text[i],  xy = (Correlation['Pearson_correlation'][i]+0.005, Correlation['Spearman_correlation'][i])\n",
    "                     , xytext = (Correlation['Pearson_correlation'][i]+.2, Correlation['Spearman_correlation'][i]-.2)\n",
    "                     , size=15, color = palette3[2],\n",
    "                    arrowprops=dict(facecolor=palette3[2]\n",
    "                                    , shrink=0.05\n",
    "                                    , alpha = .8\n",
    "                                   ))\n",
    "        \n",
    "    if text[i] == 'COL2A1':\n",
    "        plt.annotate(text[i],  xy = (Correlation['Pearson_correlation'][i]+0.005, Correlation['Spearman_correlation'][i])\n",
    "                     , xytext = (Correlation['Pearson_correlation'][i]-.3, Correlation['Spearman_correlation'][i]+.15)\n",
    "                     , size=15, color = palette3[2],\n",
    "                    arrowprops=dict(facecolor=palette3[2]\n",
    "                                    , shrink=0.05\n",
    "                                    , alpha = .8\n",
    "                                   ))\n",
    "\n",
    "plt.xlabel(\"Pearson Correlation\")\n",
    "plt.ylabel(\"Spearman Correlation\")\n",
    "\n",
    "#plt.savefig(PATH + \"Projects/SEC31A_Exon24c/Figures/SEC31A_E24c_GE_correlation_2.png\", dpi = 600, bbox_inches = \"tight\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cec4548a",
   "metadata": {},
   "outputs": [],
   "source": [
    "#preparing Splicing data and gene epxression data to be ploted in linear regression\n",
    "genecounts_all_Exon24_transpose = genecounts_all_Exon24.transpose().reset_index()\n",
    "genecounts_all_Exon24_transpose['index'] = genecounts_all_Exon24_transpose['index'].str.replace('1', '')\n",
    "genecounts_all_Exon24_transpose['index'] = genecounts_all_Exon24_transpose['index'].str.replace('2', '')\n",
    "genecounts_all_Exon24_transpose['index'] = genecounts_all_Exon24_transpose['index'].str.replace('3', '')\n",
    "genecounts_all_Exon24_transpose['index'] = genecounts_all_Exon24_transpose['index'].str.replace('4', '')\n",
    "genecounts_all_Exon24_transpose['index'] = genecounts_all_Exon24_transpose['index'].str.replace('5', '')\n",
    "genecounts_all_Exon24_transpose['index'] = genecounts_all_Exon24_transpose['index'].str.replace('6', '')\n",
    "genecounts_all_Exon24_transpose['index'] = genecounts_all_Exon24_transpose['index'].str.replace('7', '')\n",
    "genecounts_all_Exon24_transpose['index'] = genecounts_all_Exon24_transpose['index'].str.replace('8', '')\n",
    "genecounts_all_Exon24_transpose['index'] = genecounts_all_Exon24_transpose['index'].str.replace('9', '')\n",
    "genecounts_all_Exon24_transpose['index'] = genecounts_all_Exon24_transpose['index'].str.replace('0', '')\n",
    "\n",
    "genecounts_all_Exon24_transpose['index'] = genecounts_all_Exon24_transpose['index'].replace('Bonemarrow', 'bone marrow')\n",
    "genecounts_all_Exon24_transpose['index'] = genecounts_all_Exon24_transpose['index'].replace('smallintestine', 'small intestine')\n",
    "genecounts_all_Exon24_transpose['index'] = genecounts_all_Exon24_transpose['index'].replace('urinarybladder', 'urinary bladder')\n",
    "genecounts_all_Exon24_transpose['index'] = genecounts_all_Exon24_transpose['index'].replace('lymphnode', 'lymph node')\n",
    "\n",
    "list_PSI24c_mean = []\n",
    "for i in genecounts_all_Exon24_transpose.set_index('index').transpose():\n",
    "    list_PSI24c_mean.append(np.mean(genecounts_all_Exon24_transpose.set_index('index').loc[i]['SEC31A Exon 24c']))\n",
    "genecounts_all_Exon24_transpose['PSI24c_mean'] = list_PSI24c_mean\n",
    "genecounts_all_Exon24_transpose = genecounts_all_Exon24_transpose.sort_values(by = 'PSI24c_mean', ascending = False)\n",
    "genecounts_all_Exon24_transpose\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6ecb45dc",
   "metadata": {},
   "outputs": [],
   "source": [
    "#plot linear regression\n",
    "x = 'SEC31A Exon 24c'\n",
    "y = 'MIA2'\n",
    "sns.regplot(data = genecounts_all_Exon24_transpose, x = f'{x}', y = f'{y}', marker='', ci = 0, color = 'gray')\n",
    "sns.scatterplot(data = genecounts_all_Exon24_transpose, x = f'{x}', y = f'{y}', hue = 'index',  style = 'index' ,\n",
    "                palette = sns.color_palette(\"hls\", 22)).set(xlabel = 'PSI SEC31A Exon 24c', \n",
    "                                                            ylabel = 'MIA2 (normalized gene counts)')\n",
    "plt.legend(bbox_to_anchor=(1.02, 1), loc='upper left', borderaxespad=0)\n",
    "\n",
    "reg_stats = linregress(genecounts_all_Exon24_transpose.dropna()[f'{x}']\n",
    "                 , genecounts_all_Exon24_transpose.dropna()[f'{y}'])\n",
    "print(reg_stats)                                 \n",
    "#plt.savefig(PATH + \"Projects/SEC31A_Exon24c/Figures/24c_MIA2_corr.png\", dpi = 600, bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d4bd7e77",
   "metadata": {},
   "outputs": [],
   "source": [
    "#plot linear regression\n",
    "x = 'SEC31A Exon 24c'\n",
    "y = 'RBM47'\n",
    "sns.regplot(data = genecounts_all_Exon24_transpose, x = f'{x}', y = f'{y}', marker=\"\", ci = 0, color = 'gray')\n",
    "sns.scatterplot(data = genecounts_all_Exon24_transpose, x = f'{x}', y = f'{y}', hue = 'index',  style = 'index' ,\n",
    "                palette = sns.color_palette(\"hls\", 22)).set(xlabel = 'PSI SEC31A Exon 24c', \n",
    "                                                            ylabel = 'RBM47 (normalized gene counts)')\n",
    "plt.legend(bbox_to_anchor=(1.02, 1), loc='upper left', borderaxespad=0)\n",
    "\n",
    "reg_stats = linregress(genecounts_all_Exon24_transpose.dropna()[f'{x}']\n",
    "                 , genecounts_all_Exon24_transpose.dropna()[f'{y}'])\n",
    "print(reg_stats)                                 \n",
    "#plt.savefig(PATH + \"Projects/SEC31A_Exon24c/Figures/24c_RBM47_corr.png\", dpi = 600, bbox_inches = \"tight\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a4a0036a",
   "metadata": {},
   "outputs": [],
   "source": [
    "y_limit = .50\n",
    "DATA = Correlation.query('GO_term !=\"other\"').query('GO_term != \"secretion\"').query('GO_term != \"mRNA splicing\"')\n",
    "\n",
    "x_data = 'GO_term'\n",
    "y_data = 'Pearson_correlation'\n",
    "group1 = DATA.query('GO_term == \"Chylomicrons\"')[y_data]\n",
    "group2 = DATA.query('GO_term == \"Collagens\"')[y_data]\n",
    "\n",
    "\n",
    "plt.figure(figsize=(3,6))\n",
    "plot = sns.swarmplot(data = DATA, x = x_data, y = y_data,  alpha=1, color='black')\n",
    "plot = sns.boxplot(data= DATA, x = x_data, y = y_data, palette = ['#7fbc41', '#de77ae']\n",
    "           # ,capsize = 0.2,             \n",
    "           # saturation = 8,             \n",
    "           # errcolor = 'black', errwidth = 2,\n",
    "           # linewidth=2, edgecolor=\"0\"\n",
    "                  )\n",
    "\n",
    "#statistics\n",
    "p_test = statistic_f(group1, group2)\n",
    "\n",
    "#calculating Cohen´s d\n",
    "cohensd = cohens_d(group1, group2)\n",
    "\n",
    "#statistics depiction with asterisks\n",
    "asterisk_text = statistic_asterisk(p_test)\n",
    "    \n",
    "x1, x2 = 0, 1   \n",
    "y, h, col = y_limit-(y_limit*0.06), y_limit*0.015, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y, y+h, y+h, y], lw=2, c=col)\n",
    "plt.text((x1+x2)*.5, y, asterisk_text, ha='center', va='bottom', color=col, size = 20)\n",
    "plot.set_ylabel(\"Pearson correlation index to exon 24c \", fontsize=14)\n",
    "print('Pvalue: ' + str(p_test))\n",
    "print('Cohen´s d: ' + str(cohensd))\n",
    "#plt.savefig(PATH + \"/Projects/SEC31A_Exon24c/Figures/E24c_corr_APO_COL.png\", dpi=600, bbox_inches='tight')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f83cf8c0",
   "metadata": {},
   "source": [
    "# RBM47 KO RNAseq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c1a1bd04",
   "metadata": {},
   "outputs": [],
   "source": [
    "RBM47_KO_SE = pd.read_excel(PATH + '/RNAseq_Felix/RBM47/RMATS/CTRLvsRBM47KD/SE_CTRLvsRBM47KD.xlsx',\n",
    "                           index_col = [2,3,4,5,6,7,8,9,10])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ae02abae",
   "metadata": {},
   "outputs": [],
   "source": [
    "RBM47_KO_E24c = RBM47_KO_SE.loc['SEC31A', 'chr4', '-', 82830936, 82830975, 82828999, 82829058, 82842139]\n",
    "RBM47_KO_E24c = RBM47_KO_E24c.drop(['Unnamed: 0', 'GeneID', 'PValueCTRLvsRBM47KD', 'DPSI_CTRLvsRBM47KD'], axis = 1)\n",
    "RBM47_KO_E24c_plot = RBM47_KO_E24c.transpose().reset_index()\n",
    "RBM47_KO_E24c_plot = RBM47_KO_E24c_plot.replace('PSI_CTRL_1', 'Ctrl')\n",
    "RBM47_KO_E24c_plot = RBM47_KO_E24c_plot.replace('PSI_CTRL_2', 'Ctrl')\n",
    "RBM47_KO_E24c_plot = RBM47_KO_E24c_plot.replace('PSI_CTRL_3', 'Ctrl')\n",
    "RBM47_KO_E24c_plot = RBM47_KO_E24c_plot.replace('PSI_RBM47KD_1', 'RBM47 KD')\n",
    "RBM47_KO_E24c_plot = RBM47_KO_E24c_plot.replace('PSI_RBM47KD_2', 'RBM47 KD')\n",
    "RBM47_KO_E24c_plot = RBM47_KO_E24c_plot.replace('PSI_RBM47KD_3', 'RBM47 KD')\n",
    "RBM47_KO_E24c_plot.columns = ['treatment', 'PSI_E24c']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8dda9460",
   "metadata": {},
   "outputs": [],
   "source": [
    "y_limit = .40\n",
    "DATA = RBM47_KO_E24c_plot\n",
    "\n",
    "\n",
    "x_data = 'treatment'\n",
    "y_data = 'PSI_E24c'\n",
    "group1 = RBM47_KO_E24c_plot.query('treatment == \"Ctrl\"')[y_data]\n",
    "group2 = RBM47_KO_E24c_plot.query('treatment == \"RBM47 KD\"')[y_data]\n",
    "\n",
    "\n",
    "plt.figure(figsize=(3,6))\n",
    "plot = sns.swarmplot(data = DATA, x = x_data, y = y_data,  alpha=1, color='black')\n",
    "plot = sns.barplot(data= DATA, x = x_data, y = y_data, palette = ['#7fbc41', '#de77ae']\n",
    "            ,capsize = 0.2,             \n",
    "            saturation = 8,             \n",
    "            errcolor = 'black', errwidth = 2,\n",
    "            linewidth=2, edgecolor=\"0\")\n",
    "\n",
    "#statistics\n",
    "p_test = statistic_f(group1, group2)\n",
    "\n",
    "#calculating Cohen´s d\n",
    "cohensd = cohens_d(group1, group2)\n",
    "\n",
    "#statistics depiction with asterisks\n",
    "asterisk_text = statistic_asterisk(p_test)\n",
    "    \n",
    "x1, x2 = 0, 1   \n",
    "y, h, col = y_limit-(y_limit*0.06), y_limit*0.015, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y, y+h, y+h, y], lw=2, c=col)\n",
    "plt.text((x1+x2)*.5, y, asterisk_text, ha='center', va='bottom', color=col, size = 20)\n",
    "plot.set_ylabel(\"PSI SEC31A exon 24c (RNAseq)\", fontsize=14)\n",
    "print('Pvalue: ' + str(p_test))\n",
    "print('Cohen´s d: ' + str(cohensd))\n",
    "#plt.savefig(PATH + \"Projects/SEC31A_Exon24c/Figures/E24c_splicing_RBM47KD_predicted.png\", dpi=600, bbox_inches='tight')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "596eabc3",
   "metadata": {},
   "outputs": [],
   "source": [
    "genecounts_all_RBM47KD = pd.read_csv(PATH + '/RNAseq_Felix/RBM47/Deseq/dataframe_CTRLvsRBM47KD.csv', sep=',', index_col=0)\n",
    "genecounts_RBM47KD = genecounts_all_RBM47KD.loc['RBM47']\n",
    "genecounts_RBM47KD = genecounts_RBM47KD.transpose().reset_index()\n",
    "genecounts_RBM47KD = genecounts_RBM47KD.replace('CTRL_1', 'Ctrl')\n",
    "genecounts_RBM47KD = genecounts_RBM47KD.replace('CTRL_2', 'Ctrl')\n",
    "genecounts_RBM47KD = genecounts_RBM47KD.replace('CTRL_3', 'Ctrl')\n",
    "genecounts_RBM47KD = genecounts_RBM47KD.replace('RBM47KD_1', 'RBM47 KD')\n",
    "genecounts_RBM47KD = genecounts_RBM47KD.replace('RBM47KD_2', 'RBM47 KD')\n",
    "genecounts_RBM47KD = genecounts_RBM47KD.replace('RBM47KD_3', 'RBM47 KD')\n",
    "genecounts_RBM47KD.columns = ['treatment', 'GE_RBM47']\n",
    "genecounts_RBM47KD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "06882e97",
   "metadata": {},
   "outputs": [],
   "source": [
    "DATA = genecounts_RBM47KD\n",
    "y_limit = 11000\n",
    "x_data = 'treatment'\n",
    "y_data = 'GE_RBM47'\n",
    "group1 = DATA.query('treatment == \"Ctrl\"')[y_data]\n",
    "group2 = DATA.query('treatment == \"RBM47 KD\"')[y_data]\n",
    "\n",
    "\n",
    "\n",
    "plt.figure(figsize=(3,6))\n",
    "plot = sns.swarmplot(data = DATA, x = x_data, y = y_data,  alpha=1, color='black')\n",
    "plot = sns.barplot(data= DATA, x = x_data, y = y_data, palette = ['#b8e186', '#f1b6da']\n",
    "            ,capsize = 0.2,             \n",
    "            saturation = 8,             \n",
    "            errcolor = 'black', errwidth = 2,\n",
    "            linewidth=2, edgecolor=\"0\")\n",
    "#statistics\n",
    "p_test = statistic_f(group1, group2)\n",
    "\n",
    "#calculating Cohen´s d\n",
    "cohensd = cohens_d(group1, group2)\n",
    "\n",
    "#statistics depiction with asterisks\n",
    "asterisk_text = statistic_asterisk(p_test)\n",
    "    \n",
    "x1, x2 = 0, 1   \n",
    "y, h, col = y_limit-(y_limit*0.06), y_limit*0.015, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y, y+h, y+h, y], lw=2, c=col)\n",
    "plt.text((x1+x2)*.5, y, asterisk_text, ha='center', va='bottom', color=col, size = 20)\n",
    "plot.set_ylabel(\"RBM47 (normalized gene counts)\", fontsize=14)\n",
    "print('Pvalue: ' + str(p_test))\n",
    "print('Cohen´s d: ' + str(cohensd))\n",
    "#plt.savefig(PATH + \"Projects/SEC31A_Exon24c/Figures/RBM47_GE_RBM47KD_predicted.png\", dpi=600, bbox_inches='tight')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d93ed5ea",
   "metadata": {},
   "source": [
    "# RBM47 gene expression validation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3c45b39c",
   "metadata": {},
   "outputs": [],
   "source": [
    "RBM47GE_mousetissues = pd.DataFrame({'GE_RBM47' : [0.367292, 0.158769, 0.156583,\n",
    "                                                    0.17254, 0.264255,\n",
    "                                                    0.090873, 0.088083, 0.029462,\n",
    "                                                    0.092783, 0.092783, 0.096723,\n",
    "                                                    0.033609, 0.005263, 0.00879,\n",
    "                                                    0.002971, 0.001724, 0.002036\n",
    "                                                    ],\n",
    "                                           'tissue' : ['liver', 'liver', 'liver',\n",
    "                                                'kidney', 'kidney',\n",
    "                                                'spleen', 'spleen', 'spleen',\n",
    "                                                'lung', 'lung', 'lung',\n",
    "                                                'heart', 'heart','heart',\n",
    "                                                'brain', 'brain', 'brain'\n",
    "                                                    ]\n",
    "                                         })\n",
    "RBM47GE_mousetissues = RBM47GE_mousetissues.set_index('tissue')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c79b608b",
   "metadata": {},
   "outputs": [],
   "source": [
    "genecounts_all = pd.read_csv(PATH + '/RNAseq_Felix/human_tissues/Bioinfo/Deseq/dataframeall.csv', sep=',', index_col=0)\n",
    "genecounts_all_RBM47 = pd.DataFrame(genecounts_all.loc['RBM47'])\n",
    "genecounts_all_RBM47 = genecounts_all_RBM47.reset_index()\n",
    "genecounts_all_RBM47['index'] = genecounts_all_RBM47['index'].str.replace('1', '')\n",
    "genecounts_all_RBM47['index'] = genecounts_all_RBM47['index'].str.replace('2', '')\n",
    "genecounts_all_RBM47['index'] = genecounts_all_RBM47['index'].str.replace('3', '')\n",
    "genecounts_all_RBM47['index'] = genecounts_all_RBM47['index'].str.replace('4', '')\n",
    "genecounts_all_RBM47['index'] = genecounts_all_RBM47['index'].str.replace('5', '')\n",
    "genecounts_all_RBM47['index'] = genecounts_all_RBM47['index'].str.replace('6', '')\n",
    "genecounts_all_RBM47['index'] = genecounts_all_RBM47['index'].str.replace('7', '')\n",
    "genecounts_all_RBM47['index'] = genecounts_all_RBM47['index'].str.replace('8', '')\n",
    "genecounts_all_RBM47['index'] = genecounts_all_RBM47['index'].str.replace('9', '')\n",
    "genecounts_all_RBM47['index'] = genecounts_all_RBM47['index'].str.replace('0', '')\n",
    "genecounts_all_RBM47.columns = ['tissue', 'RBM47']\n",
    "genecounts_all_RBM47 = genecounts_all_RBM47.set_index('tissue')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7e4cb4ca",
   "metadata": {},
   "outputs": [],
   "source": [
    "#E24csplicing_mousetissues = E24csplicing_mousetissues.set_index('tissue') \n",
    "tissue_list = ['liver', 'kidney', 'spleen', 'lung', 'heart', 'brain']\n",
    "mouse_RBM47_mean = []\n",
    "human_RBM47_mean = []\n",
    "mouse_RBM47_std = []\n",
    "human_RBM47_std = []\n",
    "for i in tissue_list:\n",
    "    mouse_RBM47_mean.append(np.mean(RBM47GE_mousetissues.loc[i]['GE_RBM47']))\n",
    "    mouse_RBM47_std.append(np.std(RBM47GE_mousetissues.loc[i]['GE_RBM47']))\n",
    "    human_RBM47_mean.append(np.mean(genecounts_all_RBM47.loc[i]['RBM47']))\n",
    "    human_RBM47_std.append(np.std(genecounts_all_RBM47.loc[i]['RBM47']))\n",
    "    \n",
    "RBM47GE_validationplot_mouse = pd.DataFrame({'RBM47_GE_mouse_validation' : mouse_RBM47_mean,\n",
    "                                                 'mouse_std' : mouse_RBM47_std,\n",
    "                                                'tissue' : tissue_list,\n",
    "                                                 #'organism' : 'mouse' \n",
    "                                                 })    \n",
    "RBM47GE_validationplot_human = pd.DataFrame({'RBM47_GE_human_prediction' : human_RBM47_mean,\n",
    "                                                 'human_std' : human_RBM47_std,\n",
    "                                               # 'tissue' : tissue_list,\n",
    "                                                 #'organism' : 'human' \n",
    "                                                 })    \n",
    "RBM47GE_validationplot = pd.concat([RBM47GE_validationplot_human, RBM47GE_validationplot_mouse]\n",
    "                                       ,axis = 1)\n",
    "for n in range(0,6):\n",
    "    plt.errorbar(human_RBM47_mean[n], mouse_RBM47_mean[n], xerr = human_RBM47_std[n], yerr=mouse_RBM47_std[n]\n",
    "             , fmt=\"none\"\n",
    "            ,color = palette[n]\n",
    "             ,capsize = 3\n",
    "             ,alpha= .5\n",
    "            )\n",
    "\n",
    "sns.regplot(data = RBM47GE_validationplot, x = 'RBM47_GE_human_prediction'\n",
    "                , y = 'RBM47_GE_mouse_validation'\n",
    "           , marker = ''\n",
    "           , ci = 0\n",
    "           , color = 'gray'\n",
    "           ).set(xlabel = 'human RBM47 (normalized gene counts)'\n",
    "                                         , ylabel = 'mouse RBM47 gene expression (rel. to Hprt)')\n",
    "\n",
    "sns.scatterplot(data = RBM47GE_validationplot, x = 'RBM47_GE_human_prediction'\n",
    "                , y = 'RBM47_GE_mouse_validation'\n",
    "                , hue = 'tissue'\n",
    "                , palette = palette).set(xlabel = 'human RBM47 (normalized gene counts)'\n",
    "                                         , ylabel = 'mouse RBM47 gene expression (rel. to Hprt)')\n",
    "\n",
    "plt.legend(bbox_to_anchor=(1.02, 1), loc='upper left', borderaxespad=0)\n",
    "\n",
    "reg_stats = linregress(RBM47GE_validationplot['RBM47_GE_human_prediction']\n",
    "                 , RBM47GE_validationplot['RBM47_GE_mouse_validation'])\n",
    "print(reg_stats)\n",
    "#plt.savefig(PATH + \"Projects/SEC31A_Exon24c/Figures/RBM47_GE_mouse_validation_reg.png\", dpi = 600, bbox_inches = \"tight\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a7ba5772",
   "metadata": {},
   "source": [
    "# Mouse tisse RBM47 GE E24c splicing"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "39383cfc",
   "metadata": {},
   "outputs": [],
   "source": [
    "E24csplicing_mousetissues = pd.DataFrame({'PSI 24c' : [ 0.703915241, 0.718047615, 0.710920947\n",
    "                                                        , 0.432203698, 0.386756401, 0.420612286\n",
    "                                                        , 0.09155929, 0.097797553, 0.151566682\n",
    "                                                        , 0.040899208, 0.028932168, 0.000193505\n",
    "                                                        , 0.145476394, 0.183124488, 0.22928863\n",
    "                                                        , 0.020357807, 0.046647126, 0.016379024\n",
    "                                                        ],\n",
    "                                           'tissue' : ['liver', 'liver','liver',\n",
    "                                                     'kidney', 'kidney', 'kidney',\n",
    "                                                    'spleen', 'spleen', 'spleen',\n",
    "                                                    'lung', 'lung', 'lung',\n",
    "                                                    'heart', 'heart', 'heart',\n",
    "                                                    'brain', 'brain', 'brain',\n",
    "                                                    ]\n",
    "                                         })\n",
    "E24csplicing_mousetissues = E24csplicing_mousetissues.set_index('tissue')\n",
    "\n",
    "RBM47GE_mousetissues = pd.DataFrame({'GE_RBM47' : [0.367292, 0.158769, 0.156583,\n",
    "                                                    0.17254, 0.264255,\n",
    "                                                    0.090873, 0.088083, 0.029462,\n",
    "                                                    0.092783, 0.092783, 0.096723,\n",
    "                                                    0.033609, 0.005263, 0.00879,\n",
    "                                                    0.002971, 0.001724, 0.002036\n",
    "                                                    ],\n",
    "                                           'tissue' : ['liver', 'liver', 'liver',\n",
    "                                                'kidney', 'kidney',\n",
    "                                                'spleen', 'spleen', 'spleen',\n",
    "                                                'lung', 'lung', 'lung',\n",
    "                                                'heart', 'heart','heart',\n",
    "                                                'brain', 'brain', 'brain'\n",
    "                                                    ]\n",
    "                                         })\n",
    "RBM47GE_mousetissues = RBM47GE_mousetissues.set_index('tissue')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3d45832f",
   "metadata": {},
   "outputs": [],
   "source": [
    "#E24csplicing_mousetissues = E24csplicing_mousetissues.set_index('tissue') \n",
    "tissue_list = ['liver', 'kidney', 'spleen', 'lung', 'heart', 'brain']\n",
    "splicing_mean = []\n",
    "GE_mean = []\n",
    "splicing_std = []\n",
    "GE_std = []\n",
    "for i in tissue_list:\n",
    "    splicing_mean.append(np.mean(E24csplicing_mousetissues.loc[i]['PSI 24c']))\n",
    "    splicing_std.append(np.std(E24csplicing_mousetissues.loc[i]['PSI 24c']))\n",
    "    GE_mean.append(np.mean(RBM47GE_mousetissues.loc[i]['GE_RBM47']))\n",
    "    GE_std.append(np.std(RBM47GE_mousetissues.loc[i]['GE_RBM47']))\n",
    "    \n",
    "E24csplicing_validationplot_mouse = pd.DataFrame({'Mouse Exon 24c PSI' : splicing_mean,\n",
    "                                                 'splicing_std' : splicing_std,\n",
    "                                                'tissue' : tissue_list,\n",
    "                                                 #'organism' : 'mouse' \n",
    "                                                 })    \n",
    "E24csplicing_validationplot_human = pd.DataFrame({'Mouse RBM47 gene expression (rel. to Hprt)' : GE_mean,\n",
    "                                                 'GE_std' : GE_std,\n",
    "                                               # 'tissue' : tissue_list,\n",
    "                                                 #'organism' : 'human' \n",
    "                                                 })    \n",
    "E24csplicing_validationplot = pd.concat([E24csplicing_validationplot_human, E24csplicing_validationplot_mouse]\n",
    "                                       ,axis = 1)\n",
    "for n in range(0,6):\n",
    "    plt.errorbar(GE_mean[n], splicing_mean[n], xerr = GE_std[n], yerr=splicing_std[n]\n",
    "             , fmt=\"none\"\n",
    "            ,color = palette[n]\n",
    "             ,capsize = 3\n",
    "             ,alpha= .5\n",
    "            )\n",
    "#plt.scatter(splicing_mean, GE_mean)\n",
    "sns.scatterplot(data = E24csplicing_validationplot, x = 'Mouse RBM47 gene expression (rel. to Hprt)'\n",
    "                , y = 'Mouse Exon 24c PSI'\n",
    "                , hue = 'tissue'\n",
    "                , palette = palette) \n",
    "sns.regplot(data = E24csplicing_validationplot, x = 'Mouse RBM47 gene expression (rel. to Hprt)'\n",
    "                , y = 'Mouse Exon 24c PSI'\n",
    "           , marker = ''\n",
    "           , ci = 0\n",
    "           , color = 'gray'\n",
    "           )\n",
    "\n",
    "plt.legend(bbox_to_anchor=(1.02, 1), loc='upper left', borderaxespad=0)\n",
    "\n",
    "reg_stats = linregress(E24csplicing_validationplot['Mouse RBM47 gene expression (rel. to Hprt)']\n",
    "                 , E24csplicing_validationplot['Mouse Exon 24c PSI'])\n",
    "print(reg_stats)\n",
    "#plt.savefig(PATH + \"Projects/SEC31A_Exon24c/Figures/E24c_splicing_RBM47_GE_mouse.png\", dpi = 600, bbox_inches = \"tight\")\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4e7b8d0d",
   "metadata": {},
   "source": [
    "# RBM47 over expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c0675aa0",
   "metadata": {},
   "outputs": [],
   "source": [
    "E24c_splicing_RBM47_OE = pd.DataFrame({'PSI_E24c' : [0.0141585, 0.012621038,\n",
    "                                                    0.164037232, 0.086022517, 0.138930506\n",
    "                                                    ],\n",
    "                                           'treatment' : ['Hek Ctrl', 'Hek Ctrl',\n",
    "                                                'Hek RBM47 OE', 'Hek RBM47 OE', 'Hek RBM47 OE' \n",
    "                                                \n",
    "                                                \n",
    "                                                    ]\n",
    "                                         })"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d0278e06",
   "metadata": {},
   "outputs": [],
   "source": [
    "palette_generation_3 = sns.color_palette(\"RdGy_r\", 5)\n",
    "print(palette_generation_3)\n",
    "palette_generation_3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fd5bb804",
   "metadata": {},
   "outputs": [],
   "source": [
    "palette_generation_3 = sns.color_palette(\"RdGy_r\", 5)\n",
    "DATA = E24c_splicing_RBM47_OE\n",
    "y_limit = 0.18\n",
    "x_data = 'treatment'\n",
    "y_data = 'PSI_E24c'\n",
    "group1 = E24c_splicing_RBM47_OE.query('treatment == \"Hek Ctrl\"')[y_data]\n",
    "group2 = E24c_splicing_RBM47_OE.query('treatment == \"Hek RBM47 OE\"')[y_data]\n",
    "\n",
    "plt.figure(figsize=(3,6))\n",
    "plot = sns.swarmplot(data = DATA, x = x_data, y = y_data,  alpha=1, color='black')\n",
    "plot = sns.barplot(data= DATA, x = x_data, y = y_data\n",
    "            , palette = [palette_generation_3[0],\n",
    "                        palette_generation_3[4],\n",
    "                        ]\n",
    "            ,capsize = 0.2,             \n",
    "            saturation = 8,             \n",
    "            errcolor = 'black', errwidth = 2,\n",
    "            linewidth=2, edgecolor=\"0\")\n",
    "\n",
    "#statistics1\n",
    "p_ttest_1 = statistic_f(group1, group2)\n",
    "cohensd_1 = cohens_d(group1, group2)\n",
    "\n",
    "#statistics depiction with asterisks\n",
    "asterisk_text_1 = statistic_asterisk(p_ttest_1)\n",
    "x1, x2 = [0, 1]   \n",
    "y, h, col = y_limit-(y_limit*0.06), y_limit*0.015, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y, y+h, y+h, y], lw=2, c=col)\n",
    "plt.text((x1+x2)*.5, y, asterisk_text_1, ha='center', va='bottom', color=col, size = 20)\n",
    "\n",
    "#print(stat_text)\n",
    "print('Pvalue1: ' + str(p_ttest_1))\n",
    "print('Cohen´s d 1: ' + str(cohensd_1))\n",
    "\n",
    "plot.set_ylabel(\"PSI exon 24c\", fontsize=14)\n",
    "#plt.savefig(PATH + \"Users/felix/Documents/Phd/Projects/SEC31A_Exon24c/Figures/E24c_splicing_RBM47_OE.png\", dpi=600, bbox_inches='tight')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "86d32b9b",
   "metadata": {},
   "source": [
    "# Quantification RBM47 overexpression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b7fc5658",
   "metadata": {},
   "outputs": [],
   "source": [
    "GE_RBM47_OE_quantification = pd.DataFrame({'PSI_E24c' : [0.015356572,\n",
    "                                                                    0.019572882,\n",
    "                                                                    0.020545981,\n",
    "                                                                    124.9320654,\n",
    "                                                                    121.9376637,\n",
    "                                                                    96.33579183\n",
    "                                                    ],\n",
    "                                           'treatment' : ['Hek Ctrl', 'Hek Ctrl', 'Hek Ctrl',\n",
    "                                                'Hek RBM47 OE', 'Hek RBM47 OE', 'Hek RBM47 OE']\n",
    "                                         })\n",
    "\n",
    "palette_generation_3 = sns.color_palette(\"RdGy_r\", 5)\n",
    "DATA = GE_RBM47_OE_quantification\n",
    "y_limit = 140\n",
    "x_data = 'treatment'\n",
    "y_data = 'PSI_E24c'\n",
    "group1 = DATA.query('treatment == \"Hek Ctrl\"')[y_data]\n",
    "group2 = DATA.query('treatment == \"Hek RBM47 OE\"')[y_data]\n",
    "plt.figure(figsize=(3,6))\n",
    "plot = sns.swarmplot(data = DATA, x = x_data, y = y_data,  alpha=1, color='black')\n",
    "plot = sns.barplot(data= DATA, x = x_data, y = y_data\n",
    "            , palette = [palette_generation_3[0],\n",
    "                        palette_generation_3[4]]\n",
    "            ,capsize = 0.2,             \n",
    "            saturation = 8,             \n",
    "            errcolor = 'black', errwidth = 2,\n",
    "            linewidth=2, edgecolor=\"0\")\n",
    "\n",
    "#plot.set(yscale=\"log\")\n",
    "\n",
    "#statistics1\n",
    "p_ttest_1 = statistic_f(group1, group2)\n",
    "cohensd_1 = cohens_d(group1, group2)\n",
    "\n",
    "\n",
    "#statistics depiction with asterisks\n",
    "asterisk_text_1 = statistic_asterisk(p_ttest_1)\n",
    "x1, x2 = [0, 1]   \n",
    "y, h, col = y_limit-(y_limit*0.06), y_limit*0.015, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y, y+h, y+h, y], lw=2, c=col)\n",
    "plt.text((x1+x2)*.5, y, asterisk_text_1, ha='center', va='bottom', color=col, size = 20)\n",
    "\n",
    "#print(stat_text)\n",
    "print('Pvalue1: ' + str(p_ttest_1))\n",
    "print('Cohen´s d 1: ' + str(cohensd_1))\n",
    "\n",
    "plot.set_ylabel(\"RBM47 expression (rel. to GAPDH)\", fontsize=14)\n",
    "#plt.savefig(PATH + \"Users/felix/Documents/Phd/Projects/SEC31A_Exon24c/Figures/RBM47_OE_quantification_reltoGAPDH.png\", dpi=600, bbox_inches='tight')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9af509d9",
   "metadata": {},
   "source": [
    "# RBM47 OE Westernblott"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e0069bb8",
   "metadata": {},
   "outputs": [],
   "source": [
    "Westernblott_RBM47_OE_quantification = pd.DataFrame({'PSI_E24c' : [\n",
    "0.000567614,\n",
    "0.000886921,\n",
    "0.001879443,\n",
    "0.567954112,\n",
    "0.680190195,\n",
    "1.510938541\n",
    "                                                    ],\n",
    "                                           'treatment' : ['Hek Ctrl', 'Hek Ctrl', 'Hek Ctrl',\n",
    "                                                'Hek RBM47 OE', 'Hek RBM47 OE', 'Hek RBM47 OE']\n",
    "                                         })\n",
    "\n",
    "palette_generation_3 = sns.color_palette(\"RdGy_r\", 8)\n",
    "DATA = Westernblott_RBM47_OE_quantification\n",
    "y_limit = 1.8\n",
    "x_data = 'treatment'\n",
    "y_data = 'PSI_E24c'\n",
    "group1 = DATA.query('treatment == \"Hek Ctrl\"')[y_data]\n",
    "group2 = DATA.query('treatment == \"Hek RBM47 OE\"')[y_data]\n",
    "\n",
    "plt.figure(figsize=(3,6))\n",
    "plot = sns.swarmplot(data = DATA, x = x_data, y = y_data,  alpha=1, color='black')\n",
    "plot = sns.barplot(data= DATA, x = x_data, y = y_data\n",
    "            , palette = [palette_generation_3[0],\n",
    "                        palette_generation_3[4]]\n",
    "            ,capsize = 0.2,             \n",
    "            saturation = 8,             \n",
    "            errcolor = 'black', errwidth = 2,\n",
    "            linewidth=2, edgecolor=\"0\")\n",
    "\n",
    "#plot.set(yscale=\"log\")\n",
    "\n",
    "#statistics1\n",
    "p_ttest_1 = statistic_f(group1, group2)\n",
    "cohensd_1 = cohens_d(group1, group2)\n",
    "\n",
    "\n",
    "#statistics depiction with asterisks\n",
    "asterisk_text_1 = statistic_asterisk(p_ttest_1)\n",
    "x1, x2 = [0, 1]   \n",
    "y, h, col = y_limit-(y_limit*0.06), y_limit*0.015, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y, y+h, y+h, y], lw=2, c=col)\n",
    "plt.text((x1+x2)*.5, y, asterisk_text_1, ha='center', va='bottom', color=col, size = 20)\n",
    "\n",
    "\n",
    "#print(stat_text)\n",
    "print('Pvalue1: ' + str(p_ttest_1))\n",
    "print('Cohen´s d 1: ' + str(cohensd_1))\n",
    "\n",
    "plot.set_ylabel(\"RBM47-FLAG expression (rel. to GAPDH)\", fontsize=14)\n",
    "#plt.savefig(PATH + \"Projects/SEC31A_Exon24c/Figures/RBM47_OE_quantification_reltoGAPDH_westernblott.png\", dpi=600, bbox_inches='tight')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "88164975",
   "metadata": {},
   "source": [
    "# RBM47 knockdown"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8a7227a2",
   "metadata": {},
   "outputs": [],
   "source": [
    "E24c_splicing_RBM47_KO = pd.DataFrame({'PSI_E24c' : \n",
    "                                       [0.153331631, 0.161688104, 0.101034164, \n",
    "                                       0.065283243, 0.03235441, 0.044363869]\n",
    "                                      ,\n",
    "                                      'treatment' : \n",
    "                                       ['siCtrl', 'siCtrl','siCtrl',\n",
    "                                        'siRBM47', 'siRBM47', 'siRBM47',]\n",
    "                                      })\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "954d6e3a",
   "metadata": {},
   "outputs": [],
   "source": [
    "palette_generation_3 = sns.color_palette(\"RdGy_r\", 5)\n",
    "DATA = E24c_splicing_RBM47_KO\n",
    "y_limit = 0.18\n",
    "x_data = 'treatment'\n",
    "y_data = 'PSI_E24c'\n",
    "group1 = DATA.query('treatment == \"siCtrl\"')[y_data]\n",
    "group2 = DATA.query('treatment == \"siRBM47\"')[y_data]\n",
    "\n",
    "\n",
    "plt.figure(figsize=(3,6))\n",
    "plot = sns.swarmplot(data = DATA, x = x_data, y = y_data,  alpha=1, color='black')\n",
    "plot = sns.barplot(data= DATA, x = x_data, y = y_data\n",
    "            , palette = [palette_generation_3[0],\n",
    "                        palette_generation_3[4],\n",
    "                        palette_generation_3[1],\n",
    "                        palette_generation_3[3]]\n",
    "            ,capsize = 0.2,             \n",
    "            saturation = 8,             \n",
    "            errcolor = 'black', errwidth = 2,\n",
    "            linewidth=2, edgecolor=\"0\")\n",
    "\n",
    "#statistics1\n",
    "p_ttest = statistic_f(group1, group2)\n",
    "cohensd = cohens_d(group1, group2)\n",
    "\n",
    "#statistics depiction with asterisks\n",
    "asterisk_text_1 = statistic_asterisk(p_ttest)\n",
    "x1, x2 = [0, 1]   \n",
    "y, h, col = y_limit-(y_limit*0.06), y_limit*0.015, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y, y+h, y+h, y], lw=2, c=col)\n",
    "plt.text((x1+x2)*.5, y, asterisk_text_1, ha='center', va='bottom', color=col, size = 20)\n",
    "\n",
    "\n",
    "#print(stat_text)\n",
    "print('Pvalue1: ' + str(p_ttest))\n",
    "print('Cohen´s d 1: ' + str(cohensd))\n",
    "\n",
    "plot.set_ylabel(\"PSI exon 24c\", fontsize=14)\n",
    "#plt.savefig(PATH + \"Projects/SEC31A_Exon24c/Figures/E24c_splicing_RBM47_KO.png\", dpi=600, bbox_inches='tight')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f440616c",
   "metadata": {},
   "outputs": [],
   "source": [
    "RBM47_GE_RBM47_KO = pd.DataFrame({'RBM47_GE' : \n",
    "                                       [0.968312644, 0.878762273, 1.171652576,\n",
    "                                        0.263074777, 0.166494317, 0.355653641]\n",
    "                                      ,\n",
    "                                      'treatment' : \n",
    "                                       ['siCtrl', 'siCtrl','siCtrl',\n",
    "                                        'siRBM47', 'siRBM47', 'siRBM47',]\n",
    "                                      })\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "79d25de6",
   "metadata": {},
   "outputs": [],
   "source": [
    "palette_generation_3 = sns.color_palette(\"RdGy_r\", 5)\n",
    "DATA = RBM47_GE_RBM47_KO\n",
    "y_limit = 1.3\n",
    "x_data = 'treatment'\n",
    "y_data = 'RBM47_GE'\n",
    "group1 = DATA.query('treatment == \"siCtrl\"')[y_data]\n",
    "group2 = DATA.query('treatment == \"siRBM47\"')[y_data]\n",
    "\n",
    "\n",
    "plt.figure(figsize=(3,6))\n",
    "plot = sns.swarmplot(data = DATA, x = x_data, y = y_data,  alpha=1, color='black')\n",
    "plot = sns.barplot(data= DATA, x = x_data, y = y_data\n",
    "            , palette = [palette_generation_3[1],\n",
    "                        palette_generation_3[3],\n",
    "                        palette_generation_3[1],\n",
    "                        palette_generation_3[3]]\n",
    "            ,capsize = 0.2,             \n",
    "            saturation = 8,             \n",
    "            errcolor = 'black', errwidth = 2,\n",
    "            linewidth=2, edgecolor=\"0\")\n",
    "\n",
    "#statistics1\n",
    "p_ttest = statistic_f(group1, group2)\n",
    "cohensd = cohens_d(group1, group2)\n",
    "\n",
    "#statistics depiction with asterisks\n",
    "asterisk_text_1 = statistic_asterisk(p_ttest)\n",
    "x1, x2 = [0, 1]   \n",
    "y, h, col = y_limit-(y_limit*0.06), y_limit*0.015, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y, y+h, y+h, y], lw=2, c=col)\n",
    "plt.text((x1+x2)*.5, y, asterisk_text_1, ha='center', va='bottom', color=col, size = 20)\n",
    "\n",
    "\n",
    "#print(stat_text)\n",
    "print('Pvalue1: ' + str(p_ttest))\n",
    "print('Cohen´s d 1: ' + str(cohensd))\n",
    "\n",
    "plot.set_ylabel(\"RBM47 gene expresison (FoldChange)\", fontsize=14)\n",
    "#plt.savefig(PATH + \"Projects/SEC31A_Exon24c/Figures/RBM47_GE_RBM47_KO.png\", dpi=600, bbox_inches='tight')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "97fff0e0",
   "metadata": {},
   "source": [
    "# Minigene"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a5c43ca8",
   "metadata": {},
   "outputs": [],
   "source": [
    "Minigene_hek = pd.read_excel(PATH + '/Projects/SEC31A_Exon24c/SEC31AE24c_minigene_Hek_results.xlsx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "156673db",
   "metadata": {},
   "outputs": [],
   "source": [
    "Minigene_hek['sample'] = Minigene_hek['FLAG'] + ' ' + Minigene_hek['Minigene']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c59b9d00",
   "metadata": {},
   "outputs": [],
   "source": [
    "palette_generation = sns.color_palette(\"RdGy_r\", 9)\n",
    "DATA = Minigene_hek.query('sample != \"CTRL SD\"').query('sample != \"RBM47 SD\"')\n",
    "x_data = 'sample'\n",
    "y_data = 'E24 inclusion'\n",
    "group1 = DATA.query('sample == \"CTRL S\"')[y_data]\n",
    "group2 = DATA.query('sample == \"RBM47 S\"')[y_data]\n",
    "group5 = DATA.query('sample == \"CTRL L\"')[y_data]\n",
    "group6 = DATA.query('sample == \"RBM47 L\"')[y_data]\n",
    "group7 = DATA.query('sample == \"CTRL ES\"')[y_data]\n",
    "group8 = DATA.query('sample == \"RBM47 ES\"')[y_data]\n",
    "\n",
    "\n",
    "plt.figure(figsize=(8,6))\n",
    "plot = sns.swarmplot(data = DATA, x = x_data, y = y_data,  alpha=1, color='black')\n",
    "plot = sns.barplot(data= DATA, x = x_data, y = y_data\n",
    "            , palette = [palette_generation[1],\n",
    "                        palette_generation[7]]\n",
    "            ,capsize = 0.2,             \n",
    "            saturation = 8,             \n",
    "            errcolor = 'black', errwidth = 2,\n",
    "            linewidth=2, edgecolor=\"0\")\n",
    "\n",
    "#statistics1\n",
    "p_ttest1 = statistic_f(group1, group2)\n",
    "cohensd1 = cohens_d(group1, group2)\n",
    "\n",
    "#statistics depiction with asterisks\n",
    "y_limit = max(group2)+0.02\n",
    "asterisk_text_1 = statistic_asterisk(p_ttest1)\n",
    "x1, x2 = [4, 5]   \n",
    "y, h, col = y_limit-(y_limit*0.1), y_limit*0.015, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y, y+h, y+h, y], lw=2, c=col)\n",
    "plt.text((x1+x2)*.5, y, asterisk_text_1, ha='center', va='bottom', color=col, size = 20)\n",
    "\n",
    "#print(stat_text)\n",
    "print('Pvalue1: ' + str(p_ttest1))\n",
    "print('Cohen´s d 1: ' + str(cohensd1))\n",
    "\n",
    "\n",
    "p_ttest3 = statistic_f(group5, group6)\n",
    "cohensd3 = cohens_d(group5, group6)\n",
    "\n",
    "#statistics depiction with asterisks\n",
    "y_limit = max(group6)+0.03\n",
    "asterisk_text_2 = statistic_asterisk(p_ttest3)\n",
    "x1, x2 = [6, 7]   \n",
    "y, h, col = y_limit-(y_limit*0.1), y_limit*0.015, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y, y+h, y+h, y], lw=2, c=col)\n",
    "plt.text((x1+x2)*.5, y, asterisk_text_2, ha='center', va='bottom', color=col, size = 20)\n",
    "\n",
    "print('Pvalue3: ' + str(p_ttest3))\n",
    "print('Cohen´s d 3: ' + str(cohensd3))\n",
    "\n",
    "p_ttest6 = statistic_f(group2, group6)\n",
    "cohensd6 = cohens_d(group2, group6)\n",
    "\n",
    "#statistics depiction with asterisks\n",
    "y_limit=0.23\n",
    "asterisk_text_2 = statistic_asterisk(p_ttest6)\n",
    "x1, x2 = [5, 7]   \n",
    "y, h, col = y_limit-(y_limit*0.1), y_limit*0.015, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y, y+h, y+h, y], lw=2, c=col)\n",
    "plt.text((x1+x2)*.5, y+.002, asterisk_text_2, ha='center', va='bottom', color=col, size = 20)\n",
    "\n",
    "print('Pvalue6: ' + str(p_ttest6))\n",
    "print('Cohen´s d 6: ' + str(cohensd6))\n",
    "\n",
    "p_ttest7 = statistic_f(group7, group8)\n",
    "cohensd7 = cohens_d(group7, group8)\n",
    "\n",
    "#statistics depiction with asterisks\n",
    "y_limit = max(group7)+0.01\n",
    "asterisk_text_2 = statistic_asterisk(p_ttest6)\n",
    "x1, x2 = [2, 3]   \n",
    "y, h, col = y_limit-(y_limit*0.1), y_limit*0.015, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y, y+h, y+h, y], lw=2, c=col)\n",
    "plt.text((x1+x2)*.5, y, asterisk_text_2, ha='center', va='bottom', color=col, size = 20)\n",
    "\n",
    "print('Pvalue7: ' + str(p_ttest7))\n",
    "print('Cohen´s d 7: ' + str(cohensd7))\n",
    "\n",
    "plot.set_ylabel(\"PSI exon 24c\", fontsize=14)\n",
    "#plt.savefig(PATH + \"Projects/SEC31A_Exon24c/Figures/E24c_minigene_Hek.png\", dpi=600, bbox_inches='tight')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9093e9ee",
   "metadata": {},
   "source": [
    "# Caco2 differentiation and MO treatment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7fedd75c",
   "metadata": {},
   "outputs": [],
   "source": [
    "splicing_Caco2_MO_diff = pd.DataFrame({'PSI_E24c' : [0.151119692, 0.150565958, 0.13446239,\n",
    "                                                     0.542274688, 0.605173626, 0.603249424,\n",
    "                                                     0.066010369, 0.043055281, 0.073993873],\n",
    "            \n",
    "                                       'treatment' : ['MOCtrl', 'MOCtrl', 'MOCtrl',\n",
    "                                                     'diff. MOCtrl', 'diff. MOCtrl', 'diff. MOCtrl',\n",
    "                                                     'diff. MO24c', 'diff. MO24c', 'diff. MO24c']\n",
    "                                                      })"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0add48e1",
   "metadata": {},
   "outputs": [],
   "source": [
    "palette_generation = sns.color_palette(\"RdGy_r\", 9)\n",
    "DATA = splicing_Caco2_MO_diff\n",
    "y_limit = .70\n",
    "x_data = 'treatment'\n",
    "y_data = 'PSI_E24c'\n",
    "group1 = DATA.query('treatment == \"MOCtrl\"')[y_data]\n",
    "group2 = DATA.query('treatment == \"diff. MOCtrl\"')[y_data]\n",
    "group3 = DATA.query('treatment == \"diff. MO24c\"')[y_data]\n",
    "\n",
    "\n",
    "plt.figure(figsize=(4,6))\n",
    "plot = sns.swarmplot(data = DATA, x = x_data, y = y_data,  alpha=1, color='black')\n",
    "plot = sns.barplot(data= DATA, x = x_data, y = y_data\n",
    "            , palette = [palette_generation[1],\n",
    "                        palette_generation[2],\n",
    "                        palette_generation[7]]\n",
    "            ,capsize = 0.2,             \n",
    "            saturation = 8,             \n",
    "            errcolor = 'black', errwidth = 2,\n",
    "            linewidth=2, edgecolor=\"0\")\n",
    "\n",
    "#statistics1\n",
    "p_ttest1 = statistic_f(group1, group2)\n",
    "cohensd1 = cohens_d(group1, group2)\n",
    "\n",
    "p_ttest2 = statistic_f(group2, group3)\n",
    "cohensd2 = cohens_d(group2, group3)\n",
    "\n",
    "\n",
    "#statistics depiction with asterisks\n",
    "asterisk_text_1 = statistic_asterisk(p_ttest1)\n",
    "x1, x2 = [0, 0.9]   \n",
    "y, h, col = y_limit-(y_limit*0.1), y_limit*0.015, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y, y+h, y+h, y], lw=2, c=col)\n",
    "plt.text((x1+x2)*.5, y, asterisk_text_1, ha='center', va='bottom', color=col, size = 20)\n",
    "\n",
    "#statistics depiction with asterisks\n",
    "asterisk_text_2 = statistic_asterisk(p_ttest2)\n",
    "x1, x2 = [1.1, 2]   \n",
    "y, h, col = y_limit-(y_limit*0.1), y_limit*0.015, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y, y+h, y+h, y], lw=2, c=col)\n",
    "plt.text((x1+x2)*.5, y, asterisk_text_2, ha='center', va='bottom', color=col, size = 20)\n",
    "\n",
    "\n",
    "#print(stat_text)\n",
    "print('Pvalue1: ' + str(p_ttest1))\n",
    "print('Cohen´s d 1: ' + str(cohensd1))\n",
    "\n",
    "print('Pvalue2: ' + str(p_ttest2))\n",
    "print('Cohen´s d 2: ' + str(cohensd2))\n",
    "\n",
    "\n",
    "plot.set_ylabel(\"PSI exon 24c\", fontsize=14)\n",
    "#plt.savefig(PATH + \"Projects/SEC31A_Exon24c/Figures/E24c_splicing_Caco2diff_MO.png\", dpi=600, bbox_inches='tight')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "53931f88",
   "metadata": {},
   "outputs": [],
   "source": [
    "Exon24c_splicing_Caco2_MO_diff = pd.DataFrame({'Y_data' : [0.004693891,\n",
    "                                                        0.003400588,\n",
    "                                                        0.003594483,\n",
    "                                                        0.013461659,\n",
    "                                                        0.022097087,\n",
    "                                                        0.015625\n",
    "                                                     ],\n",
    "                                       'treatment' : [\n",
    "                                                      'undifferentiated', 'undifferentiated', 'undifferentiated',\n",
    "                                                     \n",
    "                                                     'differentatied', 'differentatied', 'differentatied'\n",
    "                                                     ]\n",
    "                                                      })\n",
    "\n",
    "\n",
    "palette_generation = sns.color_palette(\"RdGy_r\", 9)\n",
    "DATA = Exon24c_splicing_Caco2_MO_diff\n",
    "y_limit = 0.025\n",
    "x_data = 'treatment'\n",
    "y_data = 'Y_data'\n",
    "group1 = DATA.query('treatment == \"undifferentiated\"')[y_data]\n",
    "group2 = DATA.query('treatment == \"differentatied\"')[y_data]\n",
    "\n",
    "\n",
    "plt.figure(figsize=(3,6))\n",
    "plot = sns.swarmplot(data = DATA, x = x_data, y = y_data,  alpha=1, color='black')\n",
    "plot = sns.barplot(data= DATA, x = x_data, y = y_data\n",
    "            , palette = [palette_generation[1], palette_generation[2]]\n",
    "            ,capsize = 0.2,             \n",
    "            saturation = 8,             \n",
    "            errcolor = 'black', errwidth = 2,\n",
    "            linewidth=2, edgecolor=\"0\")\n",
    "\n",
    "#statistics1\n",
    "p_ttest1 = statistic_f(group1, group2)\n",
    "cohensd1 = cohens_d(group1, group2)\n",
    "\n",
    "\n",
    "#statistics depiction with asterisks\n",
    "asterisk_text_1 = statistic_asterisk(p_ttest1)\n",
    "x1, x2 = [0, 1]   \n",
    "y, h, col = y_limit-(y_limit*0.1), y_limit*0.015, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y, y+h, y+h, y], lw=2, c=col)\n",
    "plt.text((x1+x2)*.5, y, asterisk_text_1, ha='center', va='bottom', color=col, size = 20)\n",
    "\n",
    "#print(stat_text)\n",
    "print('Pvalue1: ' + str(p_ttest1))\n",
    "print('Cohen´s d 1: ' + str(cohensd1))\n",
    "\n",
    "\n",
    "plot.set_ylabel(\"RBM47 gene expression (rel. to GAPDH)\", fontsize=14)\n",
    "#plt.savefig(PATH + \"Projects/SEC31A_Exon24c/Figures/RBM47_GE_Caco2diff.png\", dpi=600, bbox_inches='tight')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "94314f4e",
   "metadata": {},
   "source": [
    "# Caco-2 Lipid transport assay"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1808f202",
   "metadata": {},
   "outputs": [],
   "source": [
    "NBD_basal_ratio_Caco2_MO_diff = pd.DataFrame({'y_data' : [0.387943593,\n",
    "                                                        0.418247793,\n",
    "                                                        0.473844901,\n",
    "                                                        0.242668238,\n",
    "                                                        0.173260996,\n",
    "                                                        0.253714371\n",
    "                                                     ],\n",
    "                                       'treatment' : [#'MOCtrl', 'MOCtrl', 'MOCtrl',\n",
    "                                                      'MOCtrl', 'MOCtrl', 'MOCtrl',  \n",
    "                                                     #'MO24c', 'MO24c', 'MO24c', \n",
    "                                                      'MO24c', 'MO24c',  'MO24c'\n",
    "                                                     ]\n",
    "                                                      })\n",
    "\n",
    "\n",
    "palette_generation = sns.color_palette(\"RdGy_r\", 9)\n",
    "DATA = NBD_basal_ratio_Caco2_MO_diff\n",
    "y_limit = 0.55\n",
    "x_data = 'treatment'\n",
    "y_data = 'y_data'\n",
    "group1 = DATA.query('treatment == \"MOCtrl\"')[y_data]\n",
    "group2 = DATA.query('treatment == \"MO24c\"')[y_data]\n",
    "\n",
    "\n",
    "plt.figure(figsize=(3,6))\n",
    "plot = sns.swarmplot(data = DATA, x = x_data, y = y_data,  alpha=1, color='black')\n",
    "plot = sns.barplot(data= DATA, x = x_data, y = y_data\n",
    "            , palette = [palette_generation[2],\n",
    "                        palette_generation[7]]\n",
    "            ,capsize = 0.2,             \n",
    "            saturation = 8,             \n",
    "            errcolor = 'black', errwidth = 2,\n",
    "            linewidth=2, edgecolor=\"0\")\n",
    "\n",
    "#statistics1\n",
    "p_ttest1 = statistic_f(group1, group2)\n",
    "cohensd1 = cohens_d(group1, group2)\n",
    "\n",
    "\n",
    "#statistics depiction with asterisks\n",
    "asterisk_text_1 = statistic_asterisk(p_ttest1)\n",
    "x1, x2 = [0, 1]   \n",
    "y, h, col = y_limit-(y_limit*0.1), y_limit*0.015, 'k'\n",
    "plt.plot([x1, x1, x2, x2], [y, y+h, y+h, y], lw=2, c=col)\n",
    "plt.text((x1+x2)*.5, y, asterisk_text_1, ha='center', va='bottom', color=col, size = 20)\n",
    "\n",
    "#print(stat_text)\n",
    "print('Pvalue1: ' + str(p_ttest1))\n",
    "print('Cohen´s d 1: ' + str(cohensd1))\n",
    "\n",
    "\n",
    "plot.set_ylabel(\"Fluorescence signal ratio (Basal / Apical)\", fontsize=14)\n",
    "#plt.savefig(PATH + \"Projects/SEC31A_Exon24c/Figures/Lipid_assay_rel_basal_signal.png\", dpi=600, bbox_inches='tight')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "9fefcc30",
   "metadata": {},
   "source": [
    "# Alphafold"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3fa0da7c",
   "metadata": {},
   "outputs": [],
   "source": [
    "path = PATH + '/alphafold/multimer/active_fragment/SAR1A_SEC23_SEC31A_Iso1_activeFragment_multimer/'\n",
    "file_path = glob.glob(path + '*ranking_confidence.npy') \n",
    "model_list = []\n",
    "rankingconf_list = []\n",
    "for file in file_path:\n",
    "    file_name = file.replace(path[:-1] + '\\\\' , '')\n",
    "    file_name = file_name.replace('_ranking_confidence.npy' , '')\n",
    "    file_name = file_name.replace('_multimer_v3_pred' , '')\n",
    "    Iso1_rankingconf = np.load(file)\n",
    "    model_list.append(file_name)\n",
    "    rankingconf_list.append(Iso1_rankingconf)\n",
    "Iso1_ranks = pd.DataFrame({'Models' : model_list, 'ranking confidence' : rankingconf_list})\n",
    "Iso1_ranks.sort_values(by = 'ranking confidence', ascending = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bc7641b2",
   "metadata": {},
   "outputs": [],
   "source": [
    "path = PATH + '/alphafold/multimer/active_fragment/SAR1A_SEC23_SEC31A_Iso8_activeFragment_multimer/'\n",
    "file_path = glob.glob(path + '*ranking_confidence.npy') \n",
    "model_list = []\n",
    "rankingconf_list = []\n",
    "for file in file_path:\n",
    "    file_name = file.replace(path[:-1] + '\\\\' , '')\n",
    "    file_name = file_name.replace('_ranking_confidence.npy' , '')\n",
    "    file_name = file_name.replace('_multimer_v3_pred' , '')\n",
    "    Iso8_rankingconf = np.load(file)\n",
    "    model_list.append(file_name)\n",
    "    rankingconf_list.append(Iso8_rankingconf)\n",
    "Iso8_ranks = pd.DataFrame({'Models' : model_list, 'ranking confidence' : rankingconf_list})\n",
    "Iso8_ranks.sort_values(by = 'ranking confidence', ascending = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "14234b9b",
   "metadata": {},
   "outputs": [],
   "source": [
    "path = PATH + '/alphafold/multimer/active_fragment/SAR1A_SEC23_SEC31A_Iso1_activeFragment_multimer/'\n",
    "file_path = glob.glob(path + '*plddt.npy') \n",
    "Iso1_plddt = {}\n",
    "for file in file_path:\n",
    "    file_name = file.replace(path[:-1] + '\\\\' , '')\n",
    "    file_name = file_name.replace('_plddt.npy' , '')\n",
    "    file_name = file_name.replace('result_model_' , '')\n",
    "    file_name = file_name.replace('_multimer_v3_pred' , '')\n",
    "    plddt_i = np.load(file)\n",
    "    Iso1_plddt[file_name] = plddt_i"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5057007c",
   "metadata": {},
   "outputs": [],
   "source": [
    "path = PATH + '/alphafold/multimer/active_fragment/SAR1A_SEC23_SEC31A_Iso8_activeFragment_multimer/'\n",
    "file_path = glob.glob(path + '*plddt.npy') \n",
    "Iso8_plddt = {}\n",
    "for file in file_path:\n",
    "    file_name = file.replace(path[:-1] + '\\\\' , '')\n",
    "    file_name = file_name.replace('_plddt.npy' , '')\n",
    "    file_name = file_name.replace('result_model_' , '')\n",
    "    file_name = file_name.replace('_multimer_v3_pred' , '')\n",
    "    plddt_i = np.load(file)\n",
    "    Iso8_plddt[file_name] = plddt_i"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "36d66a1d",
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.figure(figsize=(12,5))\n",
    "plt.plot(Iso1_plddt['5_0'], color = '#2ca25f', alpha = .6, label = 'Iso1')\n",
    "plt.plot(Iso8_plddt['4_2'], color = '#dd1c77', alpha = .6, label = 'Iso8')\n",
    "plt.legend(loc='lower right')\n",
    "\n",
    "\n",
    "plt.plot([0, 0], [110, 0], linewidth=2, color = 'grey')\n",
    "plt.plot([198, 198], [110, 0], linewidth=2, color = 'grey')\n",
    "plt.plot([963, 963], [110, 0], linewidth=2, color = 'grey')\n",
    "#plt.plot([1050, 1050], [110, 0], linewidth=2, color = 'grey')\n",
    "\n",
    "plt.text(60, 102, 'SAR1A', fontsize=15)\n",
    "plt.text(500, 102, 'SEC23A', fontsize=15)\n",
    "plt.text(970, 105, 'SEC31A', fontsize=10)\n",
    "plt.text(970, 101, 'active fragment', fontsize=10)\n",
    "\n",
    "plt.xlabel(\"Residue number\")\n",
    "plt.ylabel(\"pLDDT\")\n",
    "#plt.show()\n",
    "#\n",
    "#plt.savefig(PATH + \"Projects/SEC31A_Exon24c/Figures/Alphafold_SAR1_SEC23_SEC31A_activeFragment_plddt_Iso1_8.png\", dpi = 600, bbox_inches = \"tight\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ea1dbe86",
   "metadata": {},
   "outputs": [],
   "source": [
    "path = PATH + '/alphafold/multimer/active_fragment/SAR1A_SEC23_SEC31A_Iso1_activeFragment_multimer/'\n",
    "file_path = glob.glob(path + '*pae.npy') \n",
    "Iso1_pae = {}\n",
    "for file in file_path:\n",
    "    file_name = file.replace(path[:-1] + '\\\\' , '')\n",
    "    file_name = file_name.replace('_pae.npy' , '')\n",
    "    file_name = file_name.replace('result_model_' , '')\n",
    "    file_name = file_name.replace('_multimer_v3_pred' , '')\n",
    "    pae_i = np.load(file)\n",
    "    Iso1_pae[file_name] = pd.DataFrame(pae_i)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ba01dfe0",
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.figure(figsize=(10,7))\n",
    "plot = sns.heatmap(Iso1_pae['5_0'], cmap = 'PiYG_r', xticklabels=100, yticklabels=100, square = True\n",
    "              )\n",
    "#plt.xticks(ticks=np.arange(0,len(Iso8_pae['5_2']),100))\n",
    "#plt.yticks(ticks=np.arange(0,len(Iso8_pae['5_2']),100))\n",
    "#yticks = np.linspace(10,100,10)\n",
    "#ylabels = np.linspace(100,1000,10)\n",
    "\n",
    "plot.set(xlabel='Scored residue', ylabel='Aligned residue')\n",
    "#plt.savefig(PATH + \"Projects/SEC31A_Exon24c/Figures/Alphafold_SAR1_SEC23_SEC31A_activeFragment_pae_Iso1.png\", dpi = 600, bbox_inches = \"tight\")\n",
    "\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a9faa525",
   "metadata": {},
   "outputs": [],
   "source": [
    "path = PATH + '/alphafold/multimer/active_fragment/SAR1A_SEC23_SEC31A_Iso8_activeFragment_multimer/'\n",
    "file_path = glob.glob(path + '*pae.npy') \n",
    "Iso8_pae = {}\n",
    "for file in file_path:\n",
    "    file_name = file.replace(path[:-1] + '\\\\' , '')\n",
    "    file_name = file_name.replace('_pae.npy' , '')\n",
    "    file_name = file_name.replace('result_model_' , '')\n",
    "    file_name = file_name.replace('_multimer_v3_pred' , '')\n",
    "    pae_i = np.load(file)\n",
    "    Iso8_pae[file_name] = pd.DataFrame(pae_i)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9d658e53",
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.figure(figsize=(10,7))\n",
    "plot = sns.heatmap(Iso8_pae['5_2'], cmap = 'PiYG_r', xticklabels=100, yticklabels=100, square = True\n",
    "              )\n",
    "#plt.xticks(ticks=np.arange(0,len(Iso8_pae['5_2']),100))\n",
    "#plt.yticks(ticks=np.arange(0,len(Iso8_pae['5_2']),100))\n",
    "#yticks = np.linspace(10,100,10)\n",
    "#ylabels = np.linspace(100,1000,10)\n",
    "\n",
    "plot.set(xlabel='Scored residue', ylabel='Aligned residue')\n",
    "#plt.savefig(PATH + \"Projects/SEC31A_Exon24c/Figures/Alphafold_SAR1_SEC23_SEC31A_activeFragment_pae_Iso8.png\", dpi = 600, bbox_inches = \"tight\")\n",
    "\n",
    "plt.show()\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fde06ddb",
   "metadata": {},
   "outputs": [],
   "source": [
    "PAE_Iso1_plot_df = pd.DataFrame({'pae' : Iso1_pae['5_1'][963:][800]})#.reset_index()\n",
    "#PAE_Iso1_plot_df = PAE_Iso1_plot_df.drop('index', axis = 1)\n",
    "PAE_Iso1_plot_df['index'] = range(976, 1037)\n",
    "PAE_Iso1_plot_df = PAE_Iso1_plot_df.set_index('index')\n",
    "PAE_Iso1_plot_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2156ae91",
   "metadata": {},
   "outputs": [],
   "source": [
    "plt.figure(figsize=(12,5))\n",
    "plt.plot(PAE_Iso1_plot_df, color = '#4d9221', alpha = .8, label = 'Iso1')\n",
    "plt.plot(Iso8_pae['5_2'][963:975][800], color = '#c51b7d', alpha = .8, label = 'Iso8')\n",
    "plt.plot(Iso8_pae['5_2'][974:989][800], color = sns.color_palette(\"cividis\", 10)[9], alpha = 1, label = 'Exon 24c')\n",
    "plt.plot(Iso8_pae['5_2'][988:][800], color = '#c51b7d', alpha = .8)\n",
    "plt.legend(loc='lower right')\n",
    "\n",
    "\n",
    "plt.xlabel(\"Residue number\")\n",
    "plt.ylabel(\"Predicted Aligned Error\")\n",
    "\n",
    "#plt.savefig(PATH + \"Users/felix/Documents/Phd/Projects/SEC31A_Exon24c/Figures/Alphafold_pae_Iso1_8.png\", dpi = 600, bbox_inches = \"tight\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "09450e3d",
   "metadata": {},
   "source": [
    "# Polypeptide alignment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0fad4c9d",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "peptide_align = pd.DataFrame({'H. sapiens' : [10,10,10,10,10,10,10,8,8,8,4,10,10,8,10,6,4,10,8,10,10,10,10,10,10,10,10,10,10,10,10,10,10,7,6,10,10,10,10]\n",
    "                             ,'P. abelii' : [10,10,10,10,10,10,10,8,8,8,4,10,10,8,10,6,4,10,8,10,10,10,10,10,10,10,10,10,10,10,10,10,10,7,6,10,10,10,10]\n",
    "                             ,'V. vulpes' : [10,10,10,10,10,10,10,8,8,8,4,10,10,8,10,6,4,10,8,10,10,10,10,10,10,10,10,10,10,10,10,10,10,7,6,10,10,10,10]\n",
    "                             ,'M. musculus' : [10,10,10,10,10,10,10,8,8,8,4,10,10,8,10,6,4,10,8,10,10,10,10,10,10,10,10,10,10,10,10,10,10,7,6,10,10,10,10]\n",
    "                             ,'R. rattus' : [10,10,10,10,10,10,10,8,8,8,4,10,10,8,10,6,4,10,8,10,10,10,10,10,10,10,10,10,10,10,10,10,10,7,6,10,10,10,10]\n",
    "                             ,'T. s. elegans' : [10,10,10,10,10,10,10,8,8,8,4,10,10,8,10,6,4,10,8,10,10,10,10,10,10,10,10,10,10,10,10,10,10,7,6,10,10,10,10]\n",
    "                             ,'G. gallus' : [10,10,10,10,10,10,10,8,8,8,4,10,10,8,10,6,4,10,8,10,10,10,10,10,10,10,10,10,10,10,10,10,10,7,6,10,10,10,10]\n",
    "                             ,'Conservation' : [10,10,10,10,10,10,10,8,8,8,4,10,10,8,10,6,4,10,8,10,10,10,10,10,10,10,10,10,10,10,10,10,10,7,6,10,10,10,10]})\n",
    "\n",
    "plt.figure(figsize=(10,3))\n",
    "plot = sns.heatmap(peptide_align.transpose(), cmap = 'PiYG', square = False,  vmin=0, vmax=10,linewidth=.0\n",
    "                   , cbar_kws={'label': 'Conservation', 'location' : \"bottom\"}\n",
    "                   )\n",
    "for label in plot.get_yticklabels():\n",
    "    if label.get_text() != 'Conservation':\n",
    "        label.set_fontstyle(\"italic\")\n",
    "    \n",
    "sequences = [['P', 'A', 'S', 'Q', 'R', 'T', 'E', 'N', 'Q', 'S', 'I', 'Q', 'D', 'Q', 'A', 'P', 'M', 'L', 'E', 'G', 'P', 'Q', 'N', 'G', 'W', 'N', 'D', 'P', 'P', 'A', 'L', 'N', 'R', 'V', 'P', 'K', 'K', 'K', 'K'],\n",
    "['P', 'A', 'S', 'Q', 'R', 'T', 'E', 'N', 'Q', 'S', 'M', 'Q', 'D', 'Q', 'A', 'P', 'M', 'L', 'E', 'G', 'P', 'Q', 'N', 'G', 'W', 'N', 'D', 'P', 'P', 'A', 'L', 'N', 'R', 'V', 'P', 'K', 'K', 'K', 'K'],\n",
    "['P', 'A', 'S', 'Q', 'R', 'T', 'E', 'N', 'Q', 'S', 'I', 'Q', 'D', 'Q', 'A', 'P', 'V', 'L', 'E', 'G', 'P', 'Q', 'N', 'G', 'W', 'N', 'D', 'P', 'P', 'A', 'L', 'N', 'R', 'V', 'P', 'K', 'K', 'K', 'K'],\n",
    "['P', 'A', 'S', 'Q', 'R', 'T', 'E', 'N', 'Q', 'S', 'F', 'Q', 'D', 'Q', 'A', 'S', 'I', 'L', 'E', 'G', 'P', 'Q', 'N', 'G', 'W', 'N', 'D', 'P', 'P', 'A', 'L', 'N', 'R', 'V', 'P', 'K', 'K', 'K', 'K'],\n",
    "['P', 'A', 'S', 'Q', 'R', 'T', 'E', 'N', 'Q', 'S', 'F', 'Q', 'D', 'Q', 'A', 'S', 'V', 'L', 'E', 'G', 'P', 'Q', 'N', 'G', 'W', 'N', 'D', 'P', 'P', 'A', 'L', 'N', 'R', 'V', 'P', 'K', 'K', 'K', 'K'],\n",
    "['P', 'A', 'S', 'Q', 'R', 'T', 'E', 'N', 'Q', 'S', 'S', 'Q', 'D', 'K', 'A', 'S', 'T', 'L', 'E', 'G', 'P', 'Q', 'N', 'G', 'W', 'N', 'D', 'P', 'P', 'A', 'L', 'N', 'R', 'M', 'T', 'K', 'K', 'K', 'K'],\n",
    "['P', 'A', 'S', 'Q', 'R', 'T', 'E', 'K', 'P', 'P', 'A', 'Q', 'D', 'Q', 'A', 'S', 'P', 'L', 'A', 'G', 'P', 'Q', 'N', 'G', 'W', 'N', 'D', 'P', 'P', 'A', 'L', 'N', 'R', 'A', 'A', 'K', 'K', 'K', 'K']\n",
    "            , ['*','*','*','*','*','*','*','8','8','8','4','*','*','8','*','6','4','*','8','*','*','*','*','*','*','*','*','*','*','*','*','*','*','7','6','*','*','*','*']]\n",
    "y = -1\n",
    "for species in sequences:\n",
    "    x = 0\n",
    "    y = y + 1\n",
    "    for aminoacid in species:\n",
    "        \n",
    "        plot.text(x + .05, y + .8, aminoacid , size = 14.5)\n",
    "        x = x + 1\n",
    "\n",
    "#plt.savefig(PATH + \"Projects/SEC31A_Exon24c/Figures/Exon24c_conservation_long.png\", dpi = 600, bbox_inches = \"tight\")\n",
    "\n",
    "plt.show()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
